<!DOCTYPE html>
<html lang="en">

<link href="http://gmpg.org/xfn/11" rel="profile">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta http-equiv="content-type" content="text/html; charset=utf-8">
<!-- Enable responsiveness on mobile devices-->
<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">
<head>
  <title>Bradford Condon, Developer and Bioinformaticist</title>
<meta charset="utf-8">
<meta name="author" content="Bradford Condon">
<meta name="description" content="Website, CV, and Blog of Bradford Condon.  Bradford is a web and mobile developer.  Bradford also holds a PhD in plant pathology and is an expert in genomics, bioinformatics.">
</head>
<section class="hero is-primary is-bold">
    <div class="hero-body">
        <div class="container">
            <h1 class="title">
                <a href="/" title="Home">
                    Bradford Condon PhD
                </a>
            </h1>
            <h2 class="subtitle">
                Bioinformatics, Web & Mobile Development
            </h2>
        </div>

    </div>
    <div class="hero-foot">
        <nav class="tabs is-boxed nav-menu has-shadow">
            <div class="nav-center">
                <ul>

                    <li >
                    <a href="/">
          <span class="fa-stack fa-lg">
            <i class="fa fa-square-o fa-stack-2x"></i>
            <i class="fa fa-home fa-stack-1x"></i>
          </span>
                        Home</a></li>
                    <li >

                    <a href="/blog/">

          <span class="fa-stack fa-lg">
            <i class="fa fa-square-o fa-stack-2x"></i>
            <i class="fa fa-newspaper-o fa-stack-1x"></i>
          </span>
                        Blog</a>
                    </li>
                    <li >

                    <a href="/about/">
                   <span class="fa-stack fa-lg">
                      <i class="fa fa-square-o fa-stack-2x"></i>
                      <i class="fa fa-question fa-stack-1x"></i>
                    </span>
                        About</a>

                    </li>
                    <li >

                    <a href="/cv.html">
                    <span class="fa-stack fa-lg">
                        <i class="fa fa-square-o fa-stack-2x"></i>
                        <i class="fa fa-id-card fa-stack-1x"></i>
                   </span>
                        CV</a></li>
                </ul>
            </div>
        </nav>
    </div>
</section>
<link rel="stylesheet" href="/assets/css/application.css" />
<link rel="stylesheet" href="/assets/css/prism.css" />

<link rel="alternate" type="application/atom+xml" title="Bradford Condon PhD" href="/atom.xml">

<br/>

<body>
<div class="container content">
    <main>
        <div>
  <div class="box">
<article class="post">
  <h1 class="post-title">Guide to newblr output</h1>
 <time datetime="2017-07-11T00:00:00-04:00" class="post-date">11 Jul 2017</time>
  <div class="card-content">
  <p>This post is part 8 of a series on bioinformatics file formats, written for the 2017 UK-KBRIN Essentials of Next 
Generation Sequencing Workshop at the University of Kentucky.</p>

<h2 id="introduction">Introduction</h2>

<p>Newbler is a software package for <em>de novo</em> assembly of 454 sequencing data.  It is available for free from Roche as part of the GS <em>de novo</em> Assembler package.  While Roche may have discontinued support for it, you may still find yourself working with output from this assembler.</p>

<p>It outputs many files.  I’ve excerpted the description from the manual, and a table of the column labels and their meaning for each file.</p>

<h2 id="assembly-projectxml"><code class="highlighter-rouge">Assembly project.xml</code></h2>

<blockquote>
  <p>Tab-delimited file giving position-by position consensus base and flow signal information.</p>
</blockquote>

<p>We will not be going over this file, as its intended to be machine-readable.</p>

<h2 id="454alignmentinfotsv"><code class="highlighter-rouge">454AlignmentInfo.tsv</code></h2>
<blockquote>
  <p>The 454AlignmentInfo.tsv file contains position-by-position summary information about the consensus
sequence for the contigs generated by the GS De Novo Assembler application, listed one nucleotide per line (in a tabdelimited
format). Output conditionally (using the -info/-infoall/-noinfo options or the selection made on the GUI
Parameters Tab Output Sub-tab). By default, this file is only output if there are fewer than 4 million input reads and
the total length of assembled contigs is less than 40 Mbp. For larger projects, -info or -infoall or the corresponding
GUI Output selection for Alignment Info must be used to generate this file.</p>
</blockquote>

<table>
  <thead>
    <tr>
      <th>column</th>
      <th>Label</th>
      <th>Description</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>1</td>
      <td>Position</td>
      <td>Position in the contig</td>
    </tr>
    <tr>
      <td>2</td>
      <td>Consensus</td>
      <td>Consensus nucleotide at that position</td>
    </tr>
    <tr>
      <td>3</td>
      <td>Quality Score</td>
      <td>Quality score of consensus base</td>
    </tr>
    <tr>
      <td>4</td>
      <td>Unique Depth</td>
      <td>Number of unique reads at that position</td>
    </tr>
    <tr>
      <td>5</td>
      <td>Align Depth</td>
      <td>Number of reads at that position</td>
    </tr>
    <tr>
      <td>6</td>
      <td>Signal</td>
      <td>Average signal of the read flowgrams for the flows that correspond to that position</td>
    </tr>
    <tr>
      <td>7</td>
      <td>StdDeviation</td>
      <td>Standard deviation of the read flowgram signals at that position</td>
    </tr>
  </tbody>
</table>

<h2 id="454readstatustxt"><code class="highlighter-rouge">454readstatus.txt</code></h2>
<blockquote>
  <p>Tab-delimited text file providing a perread
report of the status of each read in
the assembly. The 3’ and 5’ positions of
each read’s alignment within the contig
are also reported.</p>
</blockquote>

<table>
  <thead>
    <tr>
      <th>Column</th>
      <th>Label</th>
      <th>Description</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>1</td>
      <td>Accno</td>
      <td>Accession number of the input read</td>
    </tr>
    <tr>
      <td>2</td>
      <td>Read status</td>
      <td>Status of the read in the assembly. Can be ‘assembled’, ‘partially assembled’, ‘singleton’, ‘repeat’, ‘outlier’, or ‘too short (discarded)’</td>
    </tr>
    <tr>
      <td>3</td>
      <td>5’ Contig</td>
      <td>Accession number of the contig where the 5’ end of the read aligns</td>
    </tr>
    <tr>
      <td>4</td>
      <td>5’ Position</td>
      <td>The position in the 5’ contig where the 5’ end begins</td>
    </tr>
    <tr>
      <td>5</td>
      <td>5’ Strand</td>
      <td>Orientation of the read relative to the 5’ contig.</td>
    </tr>
    <tr>
      <td>6</td>
      <td>3’ Contig</td>
      <td>Accession number of the contig where the 3’ end of the read aligns</td>
    </tr>
    <tr>
      <td>7</td>
      <td>3’ Position</td>
      <td>The position in the 5’ contig where the 5’ end begins</td>
    </tr>
    <tr>
      <td>8</td>
      <td>3’ Strand</td>
      <td>Orientation of the read relative to the 5’ contig</td>
    </tr>
  </tbody>
</table>

<h2 id="trimstatustxt"><code class="highlighter-rouge">trimstatus.txt</code></h2>
<blockquote>
  <p>Tab-delimited text file providing a perread
report of the original and revised
trim points used in the assembly.</p>
</blockquote>

<table>
  <thead>
    <tr>
      <th>Column</th>
      <th>Label</th>
      <th>Description</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>1</td>
      <td>Accno</td>
      <td>Accession number of the input read</td>
    </tr>
    <tr>
      <td>2</td>
      <td>Trimpoints used</td>
      <td>Portion of the read used (ie, 2-301)</td>
    </tr>
    <tr>
      <td>3</td>
      <td>Trimmed length</td>
      <td>Final length</td>
    </tr>
    <tr>
      <td>4</td>
      <td>Orig. trimpoints</td>
      <td>The original trimpoints specified in the input</td>
    </tr>
    <tr>
      <td>5</td>
      <td>Orig. trimmed length</td>
      <td>Original trimmed length of the read</td>
    </tr>
    <tr>
      <td>6</td>
      <td>Raw length</td>
      <td>Length of the read with no trimming</td>
    </tr>
  </tbody>
</table>

<h2 id="pairstatustxt"><code class="highlighter-rouge">pairstatus.txt</code></h2>
<blockquote>
  <p>Tab-delimited text file providing a perpair
report of the location and status of
how each paired end pair of reads were
used in the assembly</p>
</blockquote>

<table>
  <thead>
    <tr>
      <th>Column</th>
      <th>Label</th>
      <th>Description</th>
      <th> </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>1</td>
      <td>Template</td>
      <td>Template string for the pair (454 accession for 454 paired end reads)</td>
      <td> </td>
    </tr>
    <tr>
      <td>2</td>
      <td>Status</td>
      <td><code class="highlighter-rouge">Same Contig:</code> pairs are on same contig as expected.  <code class="highlighter-rouge">Link:</code> pairs are on different contigs, but could perhaps link these contigs.  All other statuses are failures.</td>
      <td> </td>
    </tr>
    <tr>
      <td>3</td>
      <td>Distance</td>
      <td>Distance between linked reads</td>
      <td> </td>
    </tr>
    <tr>
      <td>4</td>
      <td>Left contig</td>
      <td>Contig where the left half assembled</td>
      <td> </td>
    </tr>
    <tr>
      <td>5</td>
      <td>Left pos</td>
      <td>5’ position on contig</td>
      <td> </td>
    </tr>
    <tr>
      <td>6</td>
      <td>Left dir</td>
      <td>Forward + reverse -</td>
      <td> </td>
    </tr>
    <tr>
      <td>7</td>
      <td>Right contig</td>
      <td>Contig where the right half assembled</td>
      <td> </td>
    </tr>
    <tr>
      <td>8</td>
      <td>Right pos</td>
      <td>5’ position on contig</td>
      <td> </td>
    </tr>
    <tr>
      <td>9</td>
      <td>Right dir</td>
      <td>Forward + reverse -</td>
      <td> </td>
    </tr>
    <tr>
      <td>10</td>
      <td>Left distance</td>
      <td>Distance from Left pos to the end of the contig</td>
      <td> </td>
    </tr>
    <tr>
      <td>11</td>
      <td>Right distance</td>
      <td>Distance from Right pos to the end of the contig</td>
      <td> </td>
    </tr>
  </tbody>
</table>

<h2 id="454pairaligntxt"><code class="highlighter-rouge">454PairAlign.txt</code></h2>
<blockquote>
  <p>A text file giving the pairwise
alignment(s) of the overlaps used in the
assembly computation (only produced
when using the –pair option [or –pairt
option for the tab-delimited version of
the file]).</p>
</blockquote>

<table>
  <thead>
    <tr>
      <th>Column</th>
      <th>Label</th>
      <th>Description</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>1</td>
      <td>Query Accno</td>
      <td>accession number of the read</td>
    </tr>
    <tr>
      <td>2</td>
      <td>Query start</td>
      <td>Starting position of alignment in query</td>
    </tr>
    <tr>
      <td>3</td>
      <td>Query end</td>
      <td>Ending position of alignment in query</td>
    </tr>
    <tr>
      <td>4</td>
      <td>Query length</td>
      <td>Length of the query</td>
    </tr>
    <tr>
      <td>5</td>
      <td>Subj accno</td>
      <td>Accession number of subject sequence</td>
    </tr>
    <tr>
      <td>6</td>
      <td>Subj start</td>
      <td>Starting position of alignment in subject</td>
    </tr>
    <tr>
      <td>7</td>
      <td>Subj end</td>
      <td>Ending position of alignment in subject</td>
    </tr>
    <tr>
      <td>8</td>
      <td>Subj length</td>
      <td>Length of the subject</td>
    </tr>
    <tr>
      <td>9</td>
      <td>Num ident</td>
      <td>Number of matches in alignment</td>
    </tr>
    <tr>
      <td>10</td>
      <td>Align length</td>
      <td>Length of the total alignment</td>
    </tr>
    <tr>
      <td>11</td>
      <td>Query align</td>
      <td>Sequence of the query in the alignment</td>
    </tr>
    <tr>
      <td>12</td>
      <td>Subj align</td>
      <td>Sequence of the subject in the alignment</td>
    </tr>
  </tbody>
</table>

<h2 id="454newblermetricstxt"><code class="highlighter-rouge">454NewblerMetrics.txt</code></h2>
<blockquote>
  <p>File providing various assembly metrics,
including the number of input runs and
reads, the number and size of the large
consensus contigs as well as all
consensus contigs</p>
</blockquote>

<p>Please see the manual for a full description of whats included in this file.</p>

<h2 id="contigsace"><code class="highlighter-rouge">contigs.ace</code></h2>
<blockquote>
  <p>ACE format file that can be loaded by
third-party viewer programs that support
the ACE format. The output can be a
single file for the entire project or a
folder containing individual files for each
contig in the assembly</p>
</blockquote>

<p>This file is not necessarily intended to be human readable.  Please see the manual for a full description of how to interpret this file.</p>

<h2 id="contiggraphtxt"><code class="highlighter-rouge">contiggraph.txt</code></h2>

<blockquote>
  <p>The 454ContigGraph.txt file contains a graph-based description of the branching structure of an assembly’s contigs,
where nodes represent the contigs and edges between contigs give the branching structure. When paired end reads
are included in the assembly, the file also shows scaffold edges, describing how the contigs have been linked together
into scaffolds. The entries in the file are grouped by type, of which there are six, but not all projects will contain
entries for every type. Two types of entries(“S” and “P”) are only present when paired end reads are included in the
assembly.</p>
</blockquote>

<p>This file is not necessarily intended to be human readable.  Please see the manual for a full description of how to interpret this file.</p>


  </div>
</article>
  </div>



 <div class="box">
<aside class="related">
  <h3>Related posts</h3>
  <ul class="related-posts">
    
      <li>
        <a href="/2018/05/02/tripal_testing_guidelines/">
          Testing in Tripal Core
          <small><time datetime="2018-05-02T00:00:00-04:00">02 May 2018</time></small>
        </a>
      </li>
    
      <li>
        <a href="/2018/03/28/loading_gff3_tripal/">
          Tripal GFF Loader
          <small><time datetime="2018-03-28T00:00:00-04:00">28 Mar 2018</time></small>
        </a>
      </li>
    
      <li>
        <a href="/2018/03/05/tripal_guide_metadata_practices/">
          Tripal best practices
          <small><time datetime="2018-03-05T00:00:00-05:00">05 Mar 2018</time></small>
        </a>
      </li>
    
  </ul>
</aside>
 </div>


   
 </div>
    </main>
</div>

<footer class="footer">
      <div class="container">

      <div class="content has-text-centered">
      <span>
        <a href="https://github.com/bradfordcondon"><i class="fa fa-github-square fa-3x"></i></a>
        <a href="https://twitter.com/BradfordCondon"><i class="fa fa-twitter-square fa-3x"></i></a>
        <a href="https://www.linkedin.com/in/bradford-condon-0843823b/"><i class="fa fa-linkedin-square fa-3x"></i></a>

        </span>
        </div>
              <div class="content has-text-centered">

          <p>
          <time datetime="2018-05-02T10:51:36-04:00"> Bradford Condon 2018</time>. Site and content provided under a <a href="https://creativecommons.org/licenses/by-nc/2.0/"> Creative Commons non-commercial 2.0 license</a>.  Proudly hosted on Github Pages. 
          </p> 
          </div>
      
        </div>
        
      </footer>
<script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
            (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-71775012-1', 'auto');
    ga('send', 'pageview');

</script>
</body>
</html>
