<form action="/search.html" method="get">
    <label for="search-box">Search</label>
    <input type="text" id="search-box" name="query">
    <input type="submit" value="search">
</form>

<ul id="search-results"></ul>

<script>
    window.store = {
    
    "2017-07-11-newblr": {
        "title": "Guide to newblr output",
            "author": "",
            "category": "",
            "content": "This post is part 8 of a series on bioinformatics file formats, written for the 2017 UK-KBRIN Essentials of Next Generation Sequencing Workshop at the University of Kentucky.IntroductionNewbler is a software package for de novo assembly of 454 sequencing data.  It is available for free from Roche as part of the GS de novo Assembler package.  While Roche may have discontinued support for it, you may still find yourself working with output from this assembler.It outputs many files.  I’ve excerpted the description from the manual, and a table of the column labels and their meaning for each file.Assembly project.xml  Tab-delimited file giving position-by position consensus base and flow signal information.We will not be going over this file, as its intended to be machine-readable.454AlignmentInfo.tsv  The 454AlignmentInfo.tsv file contains position-by-position summary information about the consensussequence for the contigs generated by the GS De Novo Assembler application, listed one nucleotide per line (in a tabdelimitedformat). Output conditionally (using the -info/-infoall/-noinfo options or the selection made on the GUIParameters Tab Output Sub-tab). By default, this file is only output if there are fewer than 4 million input reads andthe total length of assembled contigs is less than 40 Mbp. For larger projects, -info or -infoall or the correspondingGUI Output selection for Alignment Info must be used to generate this file.            column      Label      Description                  1      Position      Position in the contig              2      Consensus      Consensus nucleotide at that position              3      Quality Score      Quality score of consensus base              4      Unique Depth      Number of unique reads at that position              5      Align Depth      Number of reads at that position              6      Signal      Average signal of the read flowgrams for the flows that correspond to that position              7      StdDeviation      Standard deviation of the read flowgram signals at that position      454readstatus.txt  Tab-delimited text file providing a perreadreport of the status of each read inthe assembly. The 3’ and 5’ positions ofeach read’s alignment within the contigare also reported.            Column      Label      Description                  1      Accno      Accession number of the input read              2      Read status      Status of the read in the assembly. Can be ‘assembled’, ‘partially assembled’, ‘singleton’, ‘repeat’, ‘outlier’, or ‘too short (discarded)’              3      5’ Contig      Accession number of the contig where the 5’ end of the read aligns              4      5’ Position      The position in the 5’ contig where the 5’ end begins              5      5’ Strand      Orientation of the read relative to the 5’ contig.              6      3’ Contig      Accession number of the contig where the 3’ end of the read aligns              7      3’ Position      The position in the 5’ contig where the 5’ end begins              8      3’ Strand      Orientation of the read relative to the 5’ contig      trimstatus.txt  Tab-delimited text file providing a perreadreport of the original and revisedtrim points used in the assembly.            Column      Label      Description                  1      Accno      Accession number of the input read              2      Trimpoints used      Portion of the read used (ie, 2-301)              3      Trimmed length      Final length              4      Orig. trimpoints      The original trimpoints specified in the input              5      Orig. trimmed length      Original trimmed length of the read              6      Raw length      Length of the read with no trimming      pairstatus.txt  Tab-delimited text file providing a perpairreport of the location and status ofhow each paired end pair of reads wereused in the assembly            Column      Label      Description                         1      Template      Template string for the pair (454 accession for 454 paired end reads)                     2      Status      Same Contig: pairs are on same contig as expected.  Link: pairs are on different contigs, but could perhaps link these contigs.  All other statuses are failures.                     3      Distance      Distance between linked reads                     4      Left contig      Contig where the left half assembled                     5      Left pos      5’ position on contig                     6      Left dir      Forward + reverse -                     7      Right contig      Contig where the right half assembled                     8      Right pos      5’ position on contig                     9      Right dir      Forward + reverse -                     10      Left distance      Distance from Left pos to the end of the contig                     11      Right distance      Distance from Right pos to the end of the contig             454PairAlign.txt  A text file giving the pairwisealignment(s) of the overlaps used in theassembly computation (only producedwhen using the –pair option [or –pairtoption for the tab-delimited version ofthe file]).            Column      Label      Description                  1      Query Accno      accession number of the read              2      Query start      Starting position of alignment in query              3      Query end      Ending position of alignment in query              4      Query length      Length of the query              5      Subj accno      Accession number of subject sequence              6      Subj start      Starting position of alignment in subject              7      Subj end      Ending position of alignment in subject              8      Subj length      Length of the subject              9      Num ident      Number of matches in alignment              10      Align length      Length of the total alignment              11      Query align      Sequence of the query in the alignment              12      Subj align      Sequence of the subject in the alignment      454NewblerMetrics.txt  File providing various assembly metrics,including the number of input runs andreads, the number and size of the largeconsensus contigs as well as allconsensus contigsPlease see the manual for a full description of whats included in this file.contigs.ace  ACE format file that can be loaded bythird-party viewer programs that supportthe ACE format. The output can be asingle file for the entire project or afolder containing individual files for eachcontig in the assemblyThis file is not necessarily intended to be human readable.  Please see the manual for a full description of how to interpret this file.contiggraph.txt  The 454ContigGraph.txt file contains a graph-based description of the branching structure of an assembly’s contigs,where nodes represent the contigs and edges between contigs give the branching structure. When paired end readsare included in the assembly, the file also shows scaffold edges, describing how the contigs have been linked togetherinto scaffolds. The entries in the file are grouped by type, of which there are six, but not all projects will containentries for every type. Two types of entries(“S” and “P”) are only present when paired end reads are included in theassembly.This file is not necessarily intended to be human readable.  Please see the manual for a full description of how to interpret this file.",
        "url": "/2017/07/11/newblr/"
    }
    ,
    
    "2017-07-09-qual-format-phred": {
        "title": "qual format",
            "author": "",
            "category": "",
            "content": "This post is part 7 of a series on file formats, written for the 2017 UK-KBRIN Essentials of Next Generation Sequencing Workshop at the University of Kentucky.qual formatA .qual file provides Phred-based quality scores for a set of sequences (often a corresponding FASTA file).  It looks something like this:&gt;my_sequence14 39 8 4 50 1 100 5 0&gt;my_sequence2 3 3 40 42 35Similar to FASTA format, each sequence is defined with a header starting with the &gt; character, and all subsequente characters after the line break describe the sequence.  Unlike FASTA, the individual nucleotides are not included in this file, only the corresponding quality Phred-format scores. At this point, all we need to understand is what are Phred quality scores.  Every nucleotide in a sequence will have a quality score associated with it.  The score Q describes the probability P that the base is called in correctly:Q = –10 log(P)Let’s look at some example Phred scores below:            Phred score      Probability incorrect      % likelihood correct call                  3      1 in 2      50%              10      1 in 10      90%              20      1 in 100      99%              30      1 in 1000      99.9%              40      1 in 10,000      99.99%              50      1 in 100,000      99.999%              60      1 in 1,000,000      99.9999%      Phred scores typically range from 2-40 with current sequencing technology.  As you can see, setting a cutoff somewhere around 30 or higher results in fairly strict confidence in invidual base calls.",
        "url": "/2017/07/09/qual_format_phred/"
    }
    ,
    
    "2017-07-02-zff": {
        "title": "ZFF annotations for SNAP",
            "author": "",
            "category": "",
            "content": "This post is part 6 of a series on file formats, written for the 2017 UK-KBRIN Essentials of Next Generation Sequencing Workshop at the University of Kentucky.SNAP and ZFFSNAP stands for Semi-HMM-based Nucleic Acid Parser, it is a general purpose gene-finding utility.The ZFF file is generated with maker2zff and is fed into the fathom commandTo read more, read the README file distributed with SNAP.  Examples are taken from https://crc.ibest.uidaho.edu/help/Applications/SNAP.html.ZFF formatGene structure files are in ZFFformat.  Confusingly, this is the file with the .ann extension.  It is a non-standard file format that resembles both FASTA and GFF.  There is both a short and long format.Short format&gt;sequence-1Einit    201    325   Y73E7A.6Eterm   2175   2319   Y73E7A.6&gt;sequence-2Einit    201    462   Y73E7A.7Exon    1803   2031   Y73E7A.7Exon    2929   3031   Y73E7A.7Exon    3467   3624   Y73E7A.7Exon    4185   4406   Y73E7A.7Eterm   5103   5280   Y73E7A.7Long format&gt;Y73E7A.6Einit    201    325   +    90   0   2   1   Y73E7A.6Eterm   2175   2319   +   295   1   0   2   Y73E7A.6&gt;Y73E7A.7Einit    201    462   +   263   0   1   1   Y73E7A.7Exon    1803   2031   +   379   2   2   0   Y73E7A.7Exon    2929   3031   +   236   1   0   0   Y73E7A.7Exon    3467   3624   +   152   0   2   0   Y73E7A.7Exon    4185   4406   +   225   1   2   2   Y73E7A.7Eterm   5103   5280   +    46   1   0   2   Y73E7A.7The below table describes each column            ID      Type      Description                  Label      Short      See zoeFeature.h for a complete list of possible values              Begin      Short      Start location              End      Short      End location              Group      Short (optional)      All exons must share a group name              Strand      Long      Plus or minus strand              Score      Long      A float type integer for score.  This score depends on the method used to generate the file.              5’-overhang      Long      Number of bp of incomplete codon 5’              3’-overhang      Long      Number of bp of incomplete codon 3’              Frame      Long      Reading frame (0, 1, 2)      ",
        "url": "/2017/07/02/ZFF/"
    }
    ,
    
    "2017-06-14-blast-format-explained": {
        "title": "BLAST output",
            "author": "",
            "category": "",
            "content": "This post is part 5 of a series on file formats, written for the 2017 UK-KBRIN Essentials of Next Generation Sequencing Workshop at the University of Kentucky.BLASTThe full BLAST manual is avialable here: https://www.ncbi.nlm.nih.gov/books/NBK279675/BLAST output formatsBLAST output options are chosen with the --outfmt flag.  For example, blastn --outfmt 0.The out format options are:0 = pairwise,1 = query-anchored showing identities,2 = query-anchored no identities,3 = flat query-anchored, show identities,4 = flat query-anchored, no identities,5 = XML Blast output,6 = tabular,7 = tabular with comment lines,8 = Text ASN.1,9 = Binary ASN.110 = Comma-separated values11 = BLAST archive format (ASN.1)The “long-form” formats, 0-4, can be good for looking at individual alignments, but they quickly become overwhelming when performing large BLAST queries.Other formats are more machine-readable, but are harder for humans to look at.  XML and ASN.1 fall in this category.  You can read more about ASN.1 format at http://www.itu.int/en/ITU-T/asn1/Pages/introduction.aspx.The best compromise between human readability and machine parsing is probably tabular or Comma-seperated format, which produces a table that can be imported into excel for manual sorting or easily parsed with a scripting langauge such as python or perl.  Let’s look at the tabular format below.outfmt 6Out format 6 outputs a single line per entry, where each column is seperated by a tab character /t.  The columns are as follows:            label      description                  qseqid      query sequence id              sseqid      subject (e.g., reference genome/hit) sequence id              pident      percentage of identical matches              length      alignment length              mismatch      number of mismatches              gapopen      number of gap openings              qstart      start of alignment in query              qend      end of alignment in query              sstart      start of alignment in subject              send      end of alignment in subject              evalue      expect value              bitscore      bit score      Remember that the query and subject ID depend on the annotations in your query and target FASTA files.  Also remember that everything after the first space in the FASTA header is considered a description.  In the below example, gene1 and gene2 would have a qseqid of genomeA in both cases.  A simple fix would be to replace spaces with underscores.&gt;genomeA gene1AAAATTTGAAAATATATATACCATTACAG&gt;genomeA gene2GGGATATAGAGGCCCCCAACCACACCCCT",
        "url": "/2017/06/14/BLAST_format_explained/"
    }
    ,
    
    "2017-06-13-velvet-out": {
        "title": "Velvet assembly output files",
            "author": "",
            "category": "",
            "content": "This post is part 4 of a series on file formats, written for the 2017 UK-KBRIN Essentials of Next Generation Sequencing Workshop at the University of Kentucky.IntroductionThe velvet manual is hosted online in wiki format.  You can find the section on output files here.  I’ll be including relevant quotes from the manual.  I also found this site helpful, and some of the examples are taken from here.Ultimately the stats files included with the assembly can be more confusing than helpful.  Don’t be too concerned if you don’t understand all of the metrics.Below is a brief guide to the files included in the output.contigs.faFrom the manual:  This fasta file contains the sequences of the contigs longer than 2k, where k is the word-length used in velveth. If you have specified a min_contig_lgth threshold, then the contigs shorter than that value are omitted.stats.txtThis file describes the nodes in the assembly and will look like the below example.            ID      lgth      out      in      long_cov      short1_cov      short1_Ocov      short2_cov      short2_Ocov      long_nb      short1_nb      short2_nb                  1      153      0      0      0.000000      118.790850      118.790850      0.000000      0.000000      0      N/A      N/A              2      55      0      0      0.000000      14.581818      14.581818      0.000000      0.000000      0      N/A      N/A      A table describing the headers is below            Column header      Full name      Description                  ID      ID      Node ID              lgth      length      length (in k-mers)              out      3’ arcs      Number of arcs 3’              in      5’ arcs      Number of arcs 5’              long_cov      coverage short 1      coverage in long reads              short1_cov      coverage short 1      coverage in short reads (including divergent sequences)              short1_Ocov      coverage short 2      coverage in short reads (conform to consensus only, strict)              short2_cov      coverage short 2      coverage in short reads (including divergent sequences)              short2_Ocov      coverage short 2      coverage in short reads (conform to consensus only, strict)              long_nb      long reads in node      number of long reads in node              short1_nb      short 1 reads in node      number of short1 reads in node              short2_nb      short 2 reads in node      number of short2 reads in node      velvet_asm.afgThis file is meant to be read by the AMOS assembly package, so we will not be concerned with it.LastGraphThis file describes the graph produced by velvet to create the assembly.There is one header line for the entire graph, which lists the number of nodes, number of sequences, and the total hash length.Then, each node has a block with the following format: NODE $NODE_ID $COV_SHORT1 $O_COV_SHORT1 $COV_SHORT2 $O_COV_SHORT2 $ENDS_OF_KMERS_OF_NODE $ENDS_OF_KMERS_OF_TWIN_NODE These should look familiar to whats in the stats.txt file.  The ends of K-mers values are the last nucleotides of the k-mers in that node.",
        "url": "/2017/06/13/velvet_out/"
    }
    ,
    
    "2017-05-24-gtf": {
        "title": "GTF (General Transfer Format)",
            "author": "Bradford Condon",
            "category": "",
            "content": "This post is part 3 of a series on file formats, written for the 2017 UK-KBRIN Essentials of Next Generation Sequencing Workshop at the University of Kentucky.  The conference website is hosted here.General transfer format (GTF), also known as General Feature Format (GFF) 2.0, is the format for transcripts in exercise 4, RNAseq.For more details, please see the ensembl guide to GFF.GTF/GFF2.0 is fairly straightforward.  Understand that is can describe a variety of sequence features, not just transcripts.  All components of a gene’s structure, such as introns, exons, and protein CDS, can be described with GTF using the correct feature type (column 3).FormatEach feature takes up one line, with 9 columns per line (plus optional track definition lines).Columns  seqname  Chromosome or scaffold name.  source Database, project, or program name.  feature  Feature type (eg Gene)  start  Start position  end  End position  score  strand + (forward) or - (reverse)  frame 0, 1, 2  attribute  Tag “value” ; tag “value”1 transcribed_unprocessed_pseudogene  gene        11869 14409 . + . gene_id \"ENSG00000223972\"; gene_name \"DDX11L1\"; gene_source \"havana\"; gene_biotype \"transcribed_unprocessed_pseudogene\"; In the above sample GTF file from ensembl, for example, the name is (chromosome) 1, the source is transcribed_unprocessed_pseudogene, the feature type is gene, the start-end is 11869-14409, there is no applicable score (.), it’s on the forward (+) strand, there is no frame data, and there are four attributes.  The first attribute, for example, is gene_name with the value DDX11L1.Track linesThe optional track lines start with the word ‘track’, followed by space-seperated key=value pairs.",
        "url": "/2017/05/24/GTF/"
    }
    ,
    
    "2017-05-14-eyhpress": {
        "title": "Expanding Your Horizons Kentucky 2017",
            "author": "",
            "category": "",
            "content": " On April 29th 2017, the Expanding Your Horizons conference came to Kentucky for the first time.  The conference was funded by an NSF EPSCoR (National Science Foundation Experimental Program to Stimulate Competitive Research for Education and Outreach)  grant which I, along with Ellen Crocker and Susan Odom, was a PI on.I’m very grateful for all of the volunteers, corporate partners, parents, teachers, and students who made this conference a success.The deprecated site is available here.About the conference  Expanding Your Horizons (EYH) is a conference for girls in middle school and their parents/ guardians. This day-long event will include hands-on activities in science, technology, engineering, and math (STEM) for girls and educational sessions for parents/ guardians. At the conference, girls will get a chance to meet STEM role models and learn more about careers in those fields. Our ultimate goal is to motivate girls to become innovative and creative thinkers ready to meet 21st Century challenges.To learn more about EYH, visit the national website.PressPost Conference  Out of the Box, Into the Future, UK Ag News, May 11 2017  “Why are you all here?” Crocker asked the participants. “We need you. If we’re going to solve the world’s problems, if we’re going to find solutions to safe drinking water, healthy air, continue being a country that is a research powerhouse, we need the best and the brightest.”  Expanding Your Horizons Introduces Middle School Girls to STEMPre-conference  STEM Conference for Middle School Girls Welcomes Participants and UK Volunteers, UKnow April 3, 2017  UK Students, Faculty Encourage Middle School Girls to ‘Expand Their Horizons’ UKnow April 25, 2017PhotosBelow are some of my favorite photos from the event.  Please follow the links for photo credits and the full albums.Undergraduate and graduate student workshop preparationPhoto credits: UKnow article by Jenny Wells jenny.wells@uky.edu    Ellen Crocker teaching the undergraduate/graduate student workshop leaders.    Students discussing the workshop they are preparing.Set 2The full album is avialable at the UkAg Flickr page, credits UK College of Agriculture.  These photos are under a CC BY-NC-ND 2.0 license.  Materials  Flyer",
        "url": "/2017/05/14/EYHpress/"
    }
    ,
    
    "2017-05-14-fastq-format-explained": {
        "title": "FASTQ format explained",
            "author": "",
            "category": "",
            "content": "This post is part 2 of a series on file formats, written for the 2017 UK-KBRIN Essentials of Next Generation Sequencing Workshop at the University of Kentucky.  The conference website is hosted here.#FastQ sequence formatFASTQ was originally developed by the Wellcome Trust Sanger Institute to bind together FASTA sequences with their respective quality data.  It is now the standard for high-throughput sequencing output.The formatFASTQ is a four-line per sequence format.  If it looks like the raw sequence of your read takes up more than four lines, you probably have word-wrapping enabled.@SEQ_IDGATTTGGGGTTCAAAGCAGTATCGATCAAATAGTAAATCCATTTGTTCAACTCACAGTTT+!''*((((***+))%%%++)(%%%%).1***-+*''))**55CCF&gt;&gt;&gt;&gt;&gt;&gt;CCCCCCC65Line 1@SEQ_ID is  analogous to the &gt;SEQ_ID line of a FASTA file.  Like a FASTA title line, everything before the space is the sequence identifier and everything after the first space is the optional sequence description.Note that different sequence sources have different standards for the description line here.Sequences off of an illumina machine will follow this format:@Instrument_name:flow_cell_lane:tile_number_in_flowcell:x-coordinate:y-coordinate#index/pair1So for example,@myInstrument:1:3:100:200#2/2 describes a read off of myInstrument.  The read was on tile number 3 of flowcell lane 1, located at X=100 Y=200, index number 2, pair #2 of a paired-end read.Note that these standards do change: for example, in Casava 1.8 extra information is added and the description is as follows:@Instrument_name:runID:flow_cell_ID:flow_cell_lane:tile_number_in_flowcell:x-coordinate:y-coordinate pair:filtered/not_filtered:index_sequence.If you need to examine the description line of your reads, make sure you know what format is being used.Line 2Line 2 is the actual sequence.Line 3Line 3 begins with the + character followed by the sequence identifier (and description).  This identifeir and description is optional: the + is not.Line 4Line 4 contains the quality values for the sequence detailed in line 2.  This line must therefore contain the exact same number of characters as line 2: each base has a corresponding sequence score.Ordered from lowest quality (!) to highest quality (~), these are the quality score characters!\"#$%&amp;'()*+,-./0123456789:;&lt;=&gt;?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~",
        "url": "/2017/05/14/fastq_format_explained/"
    }
    ,
    
    "2017-05-03-remoteblast": {
        "title": "Introduction to BLAST",
            "author": "",
            "category": "",
            "content": "UK-BLAST workshop 2016This tutorial was developed for Dr. Mark Farman at the University of Kentucky for Undergraduate students participating in UK-BLAST.  If you are trying to learn how to run applications from the command line but are feeling lost, this guide is for you.UK-BLAST is a bioinformatics working group for undergraduate researchers.  In small labs, students often tackle the bioinformatician role in with little to no technical guidance.  The group may not have anyone with a computer science background, or the task may be beyond the skills of current personnel.  In these cases, a working group headed by a technical mentor (myself) allows students to bypass some of the pitfalls and headaches associated with learning computer science all alone.What is BLAST?BLAST, or Basic Local Alignment Search Tool, is a program compares a protein or nucleotide query sequence against a database of known sequences.  BLAST has many different applications for many fields of biology.  For example, BLAST is commonly used…  To compare sequences from different genomes.  To compare similar genes within a genome.  To create phylogenetic trees.  To identify species given a DNA sample of unknown origin.  To infer function of an unknown protein.  To map DNA to a reference.  To identify repeated sequences (self BLAST).  To find conserved motifs in a protein (PSI-BLAST)When most users access BLAST, they do so via the NCBI BLAST web portal.  Here, one can paste in a sequence and search against GenBank or other comprehensive databases.  As of 2013, Genbank includes sequences from almost 260,000 formally described species.Note: for an extremely in depth summary of how BLAST works, see the NCBI course here. Be warned: there is a lot of higher math in this link.Using BLAST remotelyAccessing data: let’s go to GenBank and access some protein sequences by pointing a browser to http://www.ncbi.nlm.nih.gov/genbank/      Search for a protein you are interested in. For starters, let’s look for ABC transporters in Magnaporthe oryzae, the fungus which causes a variety of blast diseases on different “grass” plants (including wheat and rice).  Here is a top hit.  Note that the Accession number, ELQ65981.1 as well as the gene ID 440486081 both serve to uniquely identify this protein.Clicking on the FASTA link for this sequence will display the protein’s sequence in FASTA format.  Download a protein sequence and save it to its own FASTA file.Be careful copy-pasting and opening sequence files in programs like Microsoft Word - they will introduce hidden characters that will corrupt bioinformatic analyses.   It’s best to download files directory, or to use “safe” text editors like Textwrangler or SublimeText.Searching the BLAST databaseThe National Institute of Health hosts a fantastic BLAST database at their website, available at http://blast.ncbi.nlm.nih.gov/Blast.cgi Note:  if your websession fails, you can access previous results by clicking on the “Recent Results” link.      Visit the NCBI blast database.  Explore the different types of BLAST available and make sure you understand the different BLAST “flavors”    Paste or upload your FASTA file into the protein BLAST search form (careful about giving them too many options). For now, we will use the default search parameters. Make sure that the database you are querying is the non-redundant protein database.  A view of the BLAST results is provided below.The conserved domains section gives us an overview of the protein.  You can click on individual regions of the protein to learn more about each domain.  Because we searched with an ABC protein, we see an ATP-binding cassette (ABC) domain, a ABC_membrane domain, and a second ABC domain.      Click on the conserved domains of your protein.  Write down each domain name, and what function you think each domain might perform for the protein.  We will learn more about predicting protein domains, and understanding conserved domains, in a later lesson.Navigate back to the blast results page (back on your browser), and scroll down to the colored alignments section.BLAST will list results in descending order.  In the example of the first six results above, our query protein is given at the top, and each result is drawn as a series of colored boxes.  The higher the alignment score for a region, the more similar that protein is.      From which species are the top 10 hits?    How many hits retrieved have a percent identity greater than 95%?    Are the hits with low % identity still listed as ABC transporters?    Look at the header rows in the Descriptions section.  What does each column mean?  Translate each header below.  If you need help, you can look up the term in the BLAST glossary.      Max scoreTotal ScoreQuery CoverE valueIdentAccession             Storing BLAST resultsLet’s say we wanted to store the sequences of proteins retrieved in our BLAST.  To do this, scroll down to the list of hits, click “select all”, then Download, and hit “continue” with the FASTA (complete sequence) option checked.  This will download all of our hit sequences to a single FASTA file.  By selecting Text, on the other hand, we can download the BLAST output itself (the alignments, e values, etc).      Download all sequences matched by BLAST to a single FASTA file.    In your UNIX terminal, create a working directory for this tutorial.  Move your downloaded sequences to that folder (on OS X, your default download location is /Users/yourname/Downloads/) using the mv command.  There are many things we can do with this dataset. We could look for conserved domain structures in ABC proteins, align the sequences, build a phylogenetic tree, or search new genomes for more ABC transporter sequences.  Storing the data with clear names, typically with a corresponding notebook describing what the data contains and the goal of your project, is essential for keeping track of your work!In the second part of this walkthrough, we will use a single ABC sequence to identify all candidate ABC transporters in a genome.Downloading genomes from GenbankAlternatively, we may want to download an entire genome to work with later.  Typically, we want either the protein catalogue (a single FASTA file with all of the predicted amino acid sequences for a genome, corresponding to each gene) and/or the assembly (the nucleotide sequence of the entire genome, broken up into contigs or scaffolds [sets of smaller nucleotide sequence “pieces” stitched together and assembled as best as possible.  The best possible assembly will have a single scaffold for each chromosome]).Navigating from a BLAST result to a genomeFinding this raw data can sometimes be challenging.  Let’s say we’re interested in finding our top BLAST hit above in the Magnaporthe oryzae (species name) Y34 (strain name) assembly.   If we click on it’s name, this brings us to the alignment, where we can click on the sequence ID ELQ39949.1.      How many amino acids long is this protein?    If we wanted to read the article describing this sequence (and likely this genome) what journal and issue would we find it in?    What evidence on this page confirms this protein is an ABC transporter?    Hint: try clicking on the links labeled INTERPRO and CDD.  If you click on ORGANISM Magnaporthe oryzae Y34 at the top of the page, you will arrive at NCBI’s taxonomy browser.  From here we can find many useful resources, but for right now, let’s focus on the “Entrez records” on the right hand side of the screen.  We’d like to download the “Assembly” and all of the associated proteins.  Click on the “1” next to “Assembly” to go to the Entrez assembly record.We’re almost there!The assembly page has even more information about this strain’s genome sequence.  Of particular interest is the genome coverage, the sequencing technology, the total sequence length, and the number of scaffolds.  Click on the “Download the Genbank assembly” link to access the genome download page.At this point, you should be pointed at ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA_000292585.1_MoY34_2.0.  This is an FTP site, which means we could connect to this site via the terminal to download from the command line!  It may look like a lot of stuff here: if you’re confused, check the README.TXT file.  It’s common practice for people to include a README file explaining the contents of these repositories.  We want the nucleotide sequences in FASTA format and the protein sequences in FASTA format, which correspond to the files ending in protein.faa and genomic.fna.      Download the nucleotide assembly and the full protein catalogue of this genome.    In UNIX, make a folder for your whole genome assemblies.  You may like to keep one folder for proteins, and another for nucleic acids.    Create a table (for example in excel) to keep track of your downloaded genomes.  Be sure to have a column for the file name (on your system), species name, strain name, where you downloaded the data from, reference, date downloaded, and version.  If you aren’t sure of this information, it should be on the reference page.  Building a database relevant to your projectNow that you’ve downloaded a genome, you may want to have a database of genomes relevant to your own project.  To get started on this, go back to your original BLAST search (or perform a new search if you’ve lost the page).      Following the above steps, attempt to download another organism’s full nucleotide assembly and protein catalogue.    Be sure to add its information to your genome database spreadsheet.  Genbank is one of many data repositories where datasets are available.  Depending on your project, you may download assemblies from other sources.  Databases and assemblies are updated and change, which is why its important to keep track of what version you are using, and where you got it from.In the next lesson, we’ll go over installing and running BLAST locally.",
        "url": "/2017/05/03/remoteBlast/"
    }
    ,
    
    "2017-04-23-exploringpath": {
        "title": "PATH, a guide to finding one's way",
            "author": "",
            "category": "",
            "content": "UK-BLAST workshop 2016These exercises were developed for Dr. Mark Farman at the University of Kentucky for Undergraduate students participating in UK-BLAST.  If you are trying to learn how to run applications from the command line but are feeling lost, this guide is for you.UK-BLAST is a bioinformatics working group for undergraduate researchers.  In small labs, students often tackle the bioinformatician role in with little to no technical guidance.  The group may not have anyone with a computer science background, or the task may be beyond the skills of current personnel.  In these cases, a working group headed by a technical mentor (myself) allows students to bypass some of the pitfalls and headaches associated with learning computer science all alone.PATH: A guide to finding one’s wayWritten by Bradford Condon, PhD.  Farman Labratory, University of Kentucky, 2016Background: What is a PATH?In computing, a PATH is the absolute name of a file or directory, specifiying its unique location within a file system.  When you navigate to a file on your computer, locate a song in your music library, or navigate to a website, you do so using this absolute file structure.Getting startedTo begin, you must first access the command line.  On Mac, this is through the Terminal: on Windows, you need to use a program like PuTTY.Open up your Terminal (press CMD-SPACE top open spotlight and type terminal) and you should be greeted by a command prompt like this:Where are we?  What files and folders are located where we are?  The word immediately preceeding the $ symbol is the name of my current working directory.  The next section will explain some simple commands to look around your file structure.PWD: print working directoryOur first essential command is PWD.  PWD stands for print working directory.  Type in pwd and make note of where you are located.Bradfords-MacBook-Pro:Documents chet$ pwd/Users/chet/Desktop/DocumentsBradfords-MacBook-Pro:Documents chet$ This is called our starting directory or HOME directory, and is where we will begin every time we open a new terminal.This is an essential  command to learn!The dollar sign $ is called the prompt.  It lets you know that your computer is ready to receive input commands.  Notice how in the first line I typed ‘pwd’ and hit enter.  The computer processed the command and gave me an output on the second line.  The third line shows, again, my current working directory and the prompt, letting me know it is ready receive another command.The value returned by PWD on my computer was /Users/chet/Desktop/Documents.This is where my terminal is currently located, and is referred to as my working directory.  Yours will look different but it will mean the same thing: There are a series of folders within folders telling you how to find your current working directory.  on OSX, the /Users/ directory contains the home directory for all the users of that computer.  Write the full PATH pwd returns for you below.  PATH to my home directory:  /Let’s go over a few quick commands so we can practice navigating our file structure.  LS - lists files and folders in your current directory  CD - Changes directory.      MKDIR-  Creates a directory.    Try each of the above commands yourself and see what happens.  Write what happens for each in the box below.lscdmkdirWhat did mkdir do?  It gave us a usage message, which means we didn’t use the program properly.  We tried to create a directory without specifying the name of the directory.  Let’s create a directory called test:mkdir test  Try the excercises below.  Which of the above commands will let us check if our newly made directory exists? Execute this command.  What is the full path of this new folder?  Create another folder called test.  What happens?mkdir failed because we cannot have two files with the same name in the same location.If you ever need guidance figuring out how to run a program, type the program name with the help option/flag for documentation (-h after the command: see the code box below).  As you use more and more elaborate programs, this will become very useful!cd -hls -hmkdir -hCD: change directoriescd changes the directory we are in.  If we only type in  cdIt will automatically move us to our home directory.  To navigate into our new test folder, type  issue the following two commands.cd testlsthis will move us into the test folder, and list the files in this folder.  We just made it, so it’s empty.  Next, make a new folder in this directory and use ls to see if the command was successful.mkdir testlsWe can have two folders named test in separate locations, but we cannot have two folders named test in the same location.  This is because each folder has its own ABSOLUTE PATH.  Use pwd and cd to answer the question below.  What is the full PATH of the first test directory we made?  The second?  While located in the this second test folder, issue these three commands:pwdcdpwdAs you can see, the cd command returns us to our home directory when we dont supply an argument (write something after the command name).If we want to navigate back , we could typecd testcd testpwdAlternatively: we can navigate there in one jump by typingcd test/testThis command changes directory from our current directory to a folder called test, and then to the next folder inside of that.  What’s important to understand is that we can do this from anywhere on the computer by specifying the full path.Even more importantly, we can issue a command on files in other directories by giving their full PATH!  The below command would list the contents of my test folder no matter what my current working directory is.ls /Users/chet/Desktop/Documents/test/test  Try listing the contents of a folder that is not your current working directory.Relative and absolute paths  Try the following example below (on a mac).cd /usrpwd lscd /Users/_yourNameHere_/Desktop/Documents/test/testpwdNotice that by typing cd /FOLDER, we are telling cd the absolute path of a file.  before, we typed cd FOLDER (without the backslash first).  This specified the relative path.  The absolute path of a file never changes.  The relative path is really the path from our current working directory.Comprehension questions  What is the absolute path of our second test folder?  What is the relative path of our test folder from the home directory?ShortcutsBelow are two very useful shortcuts.cd ..   ##One directory up.  Very useful for quickly navigating directories.cd ../anotherfolder  ##navigates one folder backwards, and then into another folder cd ~   #~ is a shortcut to the home directory.  It's not useful here because cd also returns us to our home directory but....cd ~/Desktop/Documents\t## Very useful when specifying the ABSOLUTE PATH of a file or folder.\t## The hash symbol indicates a comment. \t# Most programming langauge ignore everything appearing in a line after a # \t#(or a different character used for this purpose): it is for human eyes to \"annotate\" code.  From Your home directory, where does cd .. take you?Command line completion/autofillThe final essential tip of this lesson is command line completion or autofill.Bradfords-MacBook-Pro:Desktop chet$ pwd/Users/chet/DesktopBradfords-MacBook-Pro:Desktop chet$ ls$RECYCLE.BINDocumentsScreen Shot 2016-02-05 at 10.44.47 AM.pngSolved Testsimages_for_PATHBradfords-MacBook-Pro:Desktop chet$ As you can see, I’m currently located in /Users/chet/Desktop.  If I’d like to move into my test folders located in documents I could type cd Documents/test/test  However, this is a confusing file path.  It’s very easy to make mistakes typing out something like this.  Command line autofill is an extremely powerful tool to get around this.  To access it, start typing your command and press tab after writing a few characters. It will automatically fill in the rest of the file name if it is unambiguous.  From your home directory, navigate back to your test folder by using autocomplete.Sometimes, we’ll have two files with similar names.  Suppose we have two folders, test1 and test2.  How can we quickly navigate into test2?      Make two similarly named directories in current folder.  Begin using autocomplete to cd into one.  What happens?        Try pressing pressing tab twice after typing in a few letters.  What happens?  We can use autocomplete not only to change directories quickly, but correctly navigate confusing directories.Final exerciseNote: do not experiment deleting, moving, or modifying files in your root directory.  Navigate to the root of your file structure by typing cd /  What files are located here?      What is the easiest way to return to your home directory?    Try navigating from your root directory to your test directory created earlier using a single command!  You can make use of the shortcuts listed above, especially tab autocomplete.Keep in mind that being familiar with the absolute PATH of files is useful for more than just navigating around your computer.  We can issue commands in folders where we are not currently located using the PATH.  This works with both relative, and absolute, paths.for example, to list the files in a new folder, we can typecd /Users/chet/Desktopmkdir test_folderls test_folderComprehension questions  Make a new test folder and navigate into it.  From here, how can you list the contents of your home directory using…  Absolute path  Relative path  How can you make a new folder in your home directory from this remote location?",
        "url": "/2017/04/23/exploringPATH/"
    }
    ,
    
    "2017-04-19-fixheadertabs": {
        "title": "fixing tabs",
            "author": "",
            "category": "",
            "content": "As you can see below, the navigation tabs I’m using at the top of my page didn’t properly update depending on what page the user is at.  While on the CV page, the CV tab should be selected, not home.  So how do we fix it?The header is defined in the /_includes folder in a hero-footer element.&lt;div class=\"hero-foot\"&gt;  &lt;nav class=\"tabs is-boxed nav-menu has-shadow\"&gt;    &lt;div class=\"nav-center\"&gt;        &lt;ul&gt;          &lt;li  class=\"is-active\"&gt;&lt;a href=\"/\"&gt;Home&lt;/a&gt;&lt;/li&gt;          &lt;li&gt;&lt;a href=\"/about/\"&gt;About&lt;/a&gt;&lt;/li&gt;          &lt;li&gt;&lt;a href=\"/cv.html\"&gt;CV&lt;/a&gt;&lt;/li&gt;        &lt;/ul&gt;      &lt;/div&gt;    &lt;/nav&gt;  &lt;/div&gt;The list element class=\"is-active\" is what tells the styling to make a particular tab active.We can test if we are on a particular page using this snippet: ``.  So, let’s replace &lt;li class=\"is-active\"&gt; with &lt;li &gt;&lt;div class=\"hero-foot\"&gt;  &lt;nav class=\"tabs is-boxed nav-menu has-shadow\"&gt;    &lt;div class=\"nav-center\"&gt;        &lt;ul&gt;          &lt;li &gt;&lt;a href=\"/\"&gt;Home&lt;/a&gt;&lt;/li&gt;          &lt;li &gt;&lt;a href=\"/about/\"&gt;About&lt;/a&gt;&lt;/li&gt;          &lt;li &gt;&lt;a href=\"/cv.html\"&gt;CV&lt;/a&gt;&lt;/li&gt;        &lt;/ul&gt;      &lt;/div&gt;    &lt;/nav&gt;  &lt;/div&gt;Reload the site and voila, our header now updates based on our page.  Liquid tags make it easy to conditionally change the class of our header tab element to active or not.If you’d like to learn more we have been working with Liquid tags.  There are a variety of handy tags to make your static page more dynamic.",
        "url": "/2017/04/19/fixHeaderTabs/"
    }
    ,
    
    "2017-04-17-floracliffupdate": {
        "title": "Floracliff mushroom workshop",
            "author": "",
            "category": "",
            "content": "I previously announced a Workshop we hosted at Floracliff on mushroom identification In October 2016.  It’s been a while but I wanted to share a little bit about the experience and post materials.First of all, a big thank you to Beverly and Josie at Floracliff.  It was a joy working with you all.  Same goes to our wonderful participants, we couldn’t have asked for a better audience.I’m sharing photos from the workshop and material below. Special thanks to Kevin Lutes for taking professional photographs and for permission to link them here.Hope to see you all again soon!Workshop photosPhotos were taken by Kevin Lutes and used with permission.  Check out the facebook album of Keivn’s photos and also find Kevin on instagram.  Group photo of those who were able to attend the day 2 hike.  We could not have asked for better students.  Dr. Bradford Condon doing his best to make sense of a field guide.  Identifying our haul.Workshop materialBelow are links to slides for presentations.  If you really need one of the other presentations let me know and I can share them.  Fungal basics  What are mushrooms?  Mushroom characteristics  Introduction to mushroom hunting  Introduction to using a mushroom ID key  Safety and mushroom poisoning  Mushroom classification",
        "url": "/2017/04/17/floraCliffUpdate/"
    }
    ,
    
    "2017-04-11-testknitr": {
        "title": "Knitting to Jekyll",
            "author": "Bradford Condon",
            "category": "",
            "content": "Knitting R Markdown to JekyllThis is a test of the Knittr package and Jekyll.It looks like the original package for using knitr with Jekyll here has fallen out of favor for Hugo, a GO-based static web-site generator.  Pretty cool, but let’s see how easy it is to knit into Jekyll without using any packages.Load in some libraries and datarequire(ggplot2)require(knitr)attach(mtcars)Do some plotshead(mtcars)##                    mpg cyl disp  hp drat    wt  qsec vs am gear carb## Mazda RX4         21.0   6  160 110 3.90 2.620 16.46  0  1    4    4## Mazda RX4 Wag     21.0   6  160 110 3.90 2.875 17.02  0  1    4    4## Datsun 710        22.8   4  108  93 3.85 2.320 18.61  1  1    4    1## Hornet 4 Drive    21.4   6  258 110 3.08 3.215 19.44  1  0    3    1## Hornet Sportabout 18.7   8  360 175 3.15 3.440 17.02  0  0    3    2## Valiant           18.1   6  225 105 2.76 3.460 20.22  1  0    3    1qplot(mpg, drat, data=mtcars, color=cyl, main=\"mpg vs drat mtcars\")Behold my beautiul plot!RenderNext we need to get our RMD to MD, with the image paths sorted.  To do this I’ll stand on the shoulders of chepec who has implemented a solution.  I created a _knitr directory in my Jekyll site, and I included a knitr folder in my assets folder.  Make sure you update the necessary paths in the render_post.r file.Now its a matter of simply sourcing the file and running KnitPost.  Make sure that your post titles match Jekyll requirements (start with YYYY-MM-DD format date).  The result is a very rapid R-studio to web post!",
        "url": "/2017/04/11/testKnitr/"
    }
    ,
    
    "2017-04-09-introtojekyll": {
        "title": "Moving to Jekyll",
            "author": "",
            "category": "",
            "content": "Switching to JekyllThis weekend I switched my site from a Wordpress site hosted by bluehost to a Jekyll site hosted on githubpages.  I’m excited about this for plenty of reasons.  Free hostings on GithubPages  Manage all aspects of the site, including layout and posts, on github  Write in Markdown  Use R plugins to script in R Markdown (I’ll share my experience with this later!)  An excuse to practice CSS and RubyIt’s pretty win-win.If you’re looking to try this out for yourself, you’ll want Ruby/RVM, Jekyll, and github set up.Migrating my old Wordpress postsBen Balter has created a Wordpress plugin that exports existing posts to Jekyll Markdown.It’s simple: download and enable the plugin, then use the export to Jekyll command on the toolbar.All of my Wordpress posts are now neatly in my _posts folder, and all associated assets are in wp-content/uploads/[date]/.  I need to cull some of the images to cut down on space, and I have to redo the formatting of my old Wordpress posts.  For example, the MD headers have more info than we would possibly care about (such as review requests) and there are some wonky tags in the pages.  All in all, though, this works extremely well.Working with draftsI suggest reading through the Jekyll docs, but the simple gist is that we’ll write our posts directly in Markdown as drafts in _drafts.  Executing jekyll serve --draftsThis post on Hongkiat’s blog pointed me to a plugin created by Jeffrey Sambells.We write posts in the _drafts folder, and mark them as date: unpublished in the header (note that having unpublished posts in _drafts folder will mess with running the –draft server.  I recommend just commenting the line out until moving to _publish)So,---layout: posttitle: \"Moving to Jekyll\"  &lt;!-- date: unpublished --&gt;---# IntroductionThis is my post that I get to write in *Markdown*!  When we’re ready to publish the post, we move it to the _publish folder (and uncomment the date).  Our plugin will fix the date and move it to the _posts folder.Configuring the site layoutHooboy.  As someone who has almost zero experience doing this, I’d recommend starting with an existing Jekyll theme if you feel overwhelmed.  If not, pick a CSS framework and mess around, there’s no better way to learn.  You can always use this site as a starting point if you’re totally lost.Start posting!I’m really hopeful that The joy of writing in Markdown (and R Markdown) will get me back in the blogging game.",
        "url": "/2017/04/09/introToJekyll/"
    }
    ,
    
    "2017-04-07-migrating-to-github": {
        "title": "Moving to Github Pages",
            "author": "",
            "category": "",
            "content": "I’m moving to a Markdown-based blog using Jekyll and github pages hosting!Please excuse the mess while I get my site in working order.  I’m hopeful that this site will be updated more frequently once I make the switch.Cheers!-Bradford",
        "url": "/2017/04/07/migrating-to-github/"
    }
    ,
    
    "outreach-science-2017-01-10-expanding-your-horizons-comes-to-kentucky": {
        "title": "Expanding Your Horizons comes to Kentucky",
            "author": "bradford.condon@gmail.com",
            "category": "",
            "content": "I’m excited to announce that, a collaboration between Dr. Ellen Crocker, Dr. Susan Odom and myself has secured an NSF EPSCoR (National Science Foundation Experimental Program to Stimulate Competitive Research) grant to bring a science conference for young girls to University of Kentucky.  The conference, an extension of the Expanding Your Horizons (EYH) network, is a great opportunity for young girls, as well as undergraduate and graduate students.  The conference will be held April 29, 2017.  To learn more, see the writeup in UKNow, or, visit the website.  We’re still collecting applications for activities- once we have our offerings together we will open registration to participants!  ",
        "url": "/outreach/science/2017/01/10/expanding-your-horizons-comes-to-kentucky/"
    }
    ,
    
    "fun-science-2016-08-25-upcoming-workshop-mushrooms-101-at-floracliff-nature-conservatory": {
        "title": "Upcoming workshop: Mushrooms 101 at Floracliff Nature Sanctuary",
            "author": "bradford.condon@gmail.com",
            "category": "",
            "content": "I’m excited to announce that my wife Dr. Ellen Crocker and I will be teaching a workshop at the Floracliff Nature Sanctuary on Friday and Saturday, October 14th-15th 2016.  I’m particularly proud to partner with Floracliff: it’s a beautiful nature preserve with very cool educational opportunities available to the public.  Ellen has taken Rob Paratley’s fern workshop and she said it was fantastic.No promises, but our last foray into the Daniel Boone National Forest was beautiful, and yielded a pound or two of some nice chanterelles.  Wild chanterelles harvested this summer in Kentucky.  Chanterelles are an example of an easy to identify edible wild mushroom.  The only poisonous look-a-like to the chanterelle is the Jack O’Lantern mushroom.  Beginning mushroom hunters will learn that chanterelles have false gills, unlike the Jack O’Lantern.  Additionaly, Jack O’Lanterns tend to grow in dense clusters, whereas chanterelles emerge distinctly.While the workshop is full, please think about signing up for the waitlist or contacting me to hear more about future mushroom forays.Participants can expect to learn the basics of fungal biology, the many ways in which fungi participate in their communities, and the principles of fungal identification.  We don’t claim to be mushroom identification experts (in fact we are routinely humbled when attending mycological forays).  However, we are trained mycologists who can teach you to appreciate the different types of fungi and the tools used to identify them.While we will talk about identification of wild mushrooms, collecting and eating wild mushrooms can be a dangerous, even deadly, past time.  In fact, Ellen has written an excellent fact sheet on the dangers of wild mushroom consumption.  I would recommend that novice mushroom hunters consult with an expert until they are extremely confident in their mushroom identification.  Even then, novice hunters should set aside 1 or 2 mushrooms if they plan to consume them.  This will be extremely useful for identifying the consumed mushroom in the event of an unintentional poisoning. A tentative itinerary can be found here.",
        "url": "/fun/science/2016/08/25/upcoming-workshop-mushrooms-101-at-floracliff-nature-conservatory/"
    }
    ,
    
    "science-2016-08-09-577": {
        "title": "Society of Postdoctoral Scholars technical officer",
            "author": "bradford.condon@gmail.com",
            "category": "",
            "content": "I am now the technical officer for the University of Kentucky Society of Postdoctoral Scholars (SOPS).If you have any questions about SOPs, especially regarding the mailing list or the website, please let me know.",
        "url": "/science/2016/08/09/577/"
    }
    ,
    
    "science-tools-2016-07-11-where-did-macqiime-go": {
        "title": "Where did MaCQIIME go?",
            "author": "bradford.condon@gmail.com",
            "category": "",
            "content": "For those of you running MacQIIME on OSX El Capitan or higher, you may (like me) find that the MacQIIME executable is no longer in your path.I was a little confused:I couldn’t find it on my system, although I still had my MacQIIME installation in the default location.  What gives?After reinstalling MacQIIME with no luck, I finally looked at the documentation (serves me right for being lazy).  Turns out that the developers are aware of this issue, and plan to fix it soon.  For now, they offer the below line of code to execute the MacQIIME shell:&lt;span class=\"hljs-keyword\"&gt;source&lt;/span&gt; &lt;span class=\"hljs-regexp\"&gt;/macqiime/&lt;/span&gt;configs&lt;span class=\"hljs-regexp\"&gt;/bash_profile.txt &lt;/span&gt;More permanent solutions are also posted on their install page.  Thanks to the Werner lab for maintaining MacQIIME.",
        "url": "/science/tools/2016/07/11/where-did-macqiime-go/"
    }
    ,
    
    "p-557": {
        "title": "Using the command line and perl to quickly trim files",
            "author": "bradford.condon@gmail.com",
            "category": "",
            "content": "Sometimes, you have a task where the goal is to remove a certain phrase, set of characters, or trailing characters within a file. This happens all the time working with FASTA files. For whatever reason, the header description might be longer than you like. Some phylogenetics programs, for example, have a character limit on headers. Failure to fix this on your own can result in clipped headers, which in turn can result in non-unique headers and failed scripts.The simple solution is to use find and replace in your text editor. This works, but you need to type in exactly what you’d like to remove: there is no pattern matching. Furthermore, what if we have 100s of files to edit? Even if there is only one file to edit, the rule of wasted work states that if you do a task once, you’re going to have to redo it with slightly modified input data. The solution: command line perl scripts. Look at the below script:for f in *.fasta; do perl -i -pe 's/_(.*)fasta//g' \"$f\"; doneWhat does it do?&lt;strong&gt;for f in *.fasta;&lt;/strong&gt; do perl -i -pe 's/_(.*)fasta//g' \"$f\"; doneThis code block says ‘for every file ending in .fasta (the asterisk * is a wildcard), we’re going to do something. Let’s refer to the file as ‘f’.do perl -i -pe 's/_(.*)fasta//g' \"$f\";This code block runs perl on our file f (which is referred to as $f). The key part is 's/_(.*)fasta//g'&lt;code&gt;. Character by character, here is what happens:&lt;/code&gt;s – this lets perl know we’re going to substitute something./_(.*)fasta/ – this is the first part of perls regexp search. We look for any characters that fall between an underscore (_) and ‘fasta’. The (.*) tells perl “any number of characters can be between the underscore and ‘fasta’ “./g Finally, this lets perl know we are doing a global search. Without the g, perl will stop at the first match. Because my fasta files have lots and lots of headers I want to trim, I search globally.Notice there are two / characters in a row after the phrase ‘fasta’. This tells perl that we are going to replace the matched phrase with nothing. We could just as easily write 's/_(.*)fasta/lizard/g' which would replace the characters between the underscore and fasta with the word ‘lizard’doneThis is the command that wraps up our shell script. It tells the loop to go back and work on the next file it can find.My hope is that even if you don’t fully understand how perl, regular expressions, or even the command line works, this walk through is enough for you to adapt my code above to accomplish what you need.",
        "url": "/?p=557"
    }
    ,
    
    "p-541": {
        "title": "Plotting a SNP map by scaffold",
            "author": "bradford.condon@gmail.com",
            "category": "",
            "content": " I wanted to plot out SNPs along a chromosome. I was starting with SNP tables that were roughly equivalent to a .bed file:scaffold00001   97859   97859scaffold00001   176752  176752Where the first column is the scaffold # in the reference, and the second and third columns are SNP locations. I adapted the R code below to print out SNP densities for select scaffolds. Because we’re working with hundreds, if not thousands, of scaffolds, it’s necessary to filter out the scaffolds of interest in some way, either by number of SNPs or the specific scaffolds of interest. We do two quick things to help us out.  The first is use count() and match() to generate a column which essentially counts the number of SNPs for each scaffold.  Then we use order() to sort by this new SNP count column.    library(plyr)    library(ggplot2)        setwd(\"/your path\")    snps&lt;-read.table(\"out.txt\",sep=\"\\t\",header=F)        colnames(snps)&lt;-c(\"chr\",\"start\",\"end\")        #Column of # SNPs    counts= count(snps[1])    snps$Count &lt;- counts[[2]][ match( snps$chr, counts[[1]] ) ]        snpsorder= snps[order(-snps[,4]),]         subset= subset(snpsorder, Count &gt; 40000)        # Plot SNP density    toprint&lt;-ggplot(subset) +       geom_histogram(aes(x=start), binwidth=1000) + # binwidth will effectively set the printing window size.      facet_wrap(~ chr,ncol=2) + # fiddle with number of columns depending on how many scaffolds you print      ggtitle(\"SNPs\") +        xlab(\"nucleotide number\") +   ylab(\"SNP density\") +   theme_bw()             # save to .png    png(\"snps.png\",750,750)    print(toprint)    dev.off()The above R code will plot the only scaffold with over 40k SNP counts. I chose this number because it prints the below plot.If I reduce the minimum to 20,000, I print a few more scaffoldsHope this code is helpful to you! ",
        "url": "/?p=541"
    }
    ,
    
    "p-501": {
        "title": "Metagenomics in R: Phyloseq",
            "author": "bradford.condon@gmail.com",
            "category": "",
            "content": "Phyloseq is a fantastic R package that really helped me get started in R.  If you need to process your metagenomic data but lack the experience in statistics/R to use, say, the ecology stats R package VEGAN, I highly recommend giving Phyloseq a try.  Running through the vignettes will produce great heatmaps, taxonomy plots, and PCA/NMDS plots. As I became more comfortable in R, I found myself relying on Phyloseq less and using the VEGAN packages Phyloseq was wrapping more.  Still, Phyloseq is a fantastic stepping stone for scientists studying metagenomics who are not ecologists themselves.I’ll be posting some solutions to problems I’ve encountered working with my own data over the next few months. Official Phyloseq publication:  DOI: 10.1371/journal.pone.0061217  ",
        "url": "/?p=501"
    }
    ,
    
    "p-505": {
        "title": "Pardon my mess",
            "author": "bradford.condon@gmail.com",
            "category": "",
            "content": "Hello, thanks for finding my page!I’m currently experimenting with site layout and design.  Things will keep changing while I get the site how I want it.  Feel free to let me know what you think. -Bradford",
        "url": "/?p=505"
    }
    ,
    
    "p-429": {
        "title": "Extracting MinION fastq on the command line using poRe",
            "author": "BradfordCondon",
            "category": "",
            "content": "Source: Extracting MinION fastq on the command line using poRe",
        "url": "/?p=429"
    }
    ,
    
    "p-416": {
        "title": "An unfortunate sweet tooth for sensationalism",
            "author": "BradfordCondon",
            "category": "",
            "content": "When I taught a science for non-science majors at Cornell University, my favorite lecture was on science in the popular media.  My learning objective was straight forward: I wanted students to pick up a newspaper, flip to the science section, and detect bullshit.This is such an important life skill because at this point in time, blogs, newspapers, and magazines print unscientific bullshit.  They do it because it sells.  I have never seen an article so neatly package all of the problems that lead to this state of affairs than John Bohannon’s recent “Chocolate Diet” charade.  If you haven’t read it yet, please do.  I’ll wait.Amazing, right?The studyHere’s the breakdown: Diana Löbl and Peter Onneken are working on a “diet science” documentary, and want to show how diet news can get reported.  So they enlist Johannes to conduct a shoddy study.  Participants will either make no dietary changes, go on a diet, or go on a diet with a bar of chocolate.  By only including 15 people in their study, and measuring a wealth of factors, they were guaranteed to find that the chocolate dieters fared better in some way than the other groups.  The finding would be meaningless, but it would be “statistically significant” – within the confines of their prohibitively small study.  They lucked out, and the 5 chocolate eaters lost weight.  How to Lie With Statistics, by Daryl Huff.Next, they needed to publish their study somewhere.  Fortunately for Johannes, it’s getting harder and harder to distinguish “fake” publishers from real ones.  Real journals will conduct real peer review, where third-party scientists must evaluate the article before it can be published.  Due to the intentionally poor design of this study, it (hopefully) wouldn’t pass peer review in any credible journal.A fake publisher, on the other hand, will not conduct review.  Instead, it will collect a publication fee, then immediately accept and publish the article without review.  Bohannon’s paper was easily accepted into the International Archives of Medicine without any peer review.Once the paper was published, it took off in the science and diet sections of many, many magazines, blogs, and newspapers.  You can easily understand why- “Chocolate helps you lose weight” is the sort of headline The Daily Mail and its ilk dream about.The AssignmentBack to my students.  To consider a news piece credible, students first needed to ask three questions.1.  Is the finding reported by a scientist?  Who?  Does the scientist work for a biased source?2.  Was the finding published in a peer-reviewed journal?3.  Can you find the article?  Do the conclusions the authors draw in the abstract reflect what is reported in the newspaper?How would my students fare if they were given this article?  It turns out, not so well.  Here are my answers, working from Bohannon’s press release.Question 1:  Is the finding reported by a scientist?  Does the scientist work for a biased source?  Johannes Bohannon is presented as the research director of the nonprofit Institute of Diet and Health, a fictional institute.  They have a website that is intentionally minimalist.  It doesn&#8217;t look incredibly legitimate, but it also doesn&#8217;t scream crazy.  I imagine most students would give it a pass, despite the possibility that, say, this fictional institute is funded by Nestle.Question 2:  Was the finding published in a peer-reviewed journal?  Yes of course it was!  Here&#8217;s the real problem predatory journals present to the common reader going forward.  The International Archives of Medicine (IAM) is a legitimate enough sounding journal.  The Journal&#8217;s website has an ISSN, citation metrics, and assurances that it is included in the JournalGuide whitelist of reputable titles.  As Bohannon reports, this journal used to be reputable, as it was published under BioMed Central, a trustworthy open-access publisher.  In short, one would need to do some serious digging to realize the journal is bogus.Question 3.  Can you find the article?  Do the conclusions the authors draw in the abstract reflect what is reported in the newspaper?  Yes and yes.  This is an easy one, because Bohannon&#8217;s wrote both the press release and the article.The solutionIt’s clear that we have a problem here.  What prevents someone from doing poor work (or fabricating work) and “publishing” the work in predatory journals?  Nothing.  And unless some sort of watchdog agency is set up to accredit peer-reviewed journals, this will not change any time soon.  If real scientists can’t distinguish a quality journal from a predatory one, what chance do non-scientists have?  As it is, I find Beall’s list of questionable publishers to be the most useful resource for investigating journals.  But this is a list built by a science librarian who saw a problem, and can’t be expected to be definitive or exhaustive.  The JournalGuide, cited on IAM’s website, appears to be a reasonable resource in its own right (it’s listed in this Nature article on resources for evaluating journals), but obviously it’s not perfect if IAM is listed.If policing publishers isn’t the answer, then what can be done?  Bohannon’s piece makes one thing abundantly clear: the popular media did not do their homework on this piece.  Many magazines ran the story without contacting Bohannon at all.  If he was contacted, he was never contacted by someone who knew enough about science to ask the right questions.  In short, science journalism lack skepticism.  It seems that most magazines will run an article without asking the three questions I listed above.  And even if they do, it’s clear that a more careful tact must be taken in the future to prevent spreading misinformation.Until that day arrives, I urge you, the reader, to bring a healthy dose of skepticism to every new miracle finding you read, especially related to human health, sexuality, and diet.  Do your homework, because no one else is doing it for you.",
        "url": "/?p=416"
    }
    ,
    
    "p-402": {
        "title": "Gamification for positive habits",
            "author": "BradfordCondon",
            "category": "",
            "content": "  For as long as I can remember, I&#8217;ve enjoyed video games.  If there&#8217;s leveling up, character optimization, or strategic battles to be had, I&#8217;m in.  At my worst, I can burn an entire Sunday; I&#8217;ll admit that I&#8217;ve banned myself from playing DOTA2 after one too many late nights.  At their best, video games can be a vehicle for me to work on my goals.  For example, I&#8217;ve been changing the language settings on my videogames to Spanish.  I&#8217;m not learning a ton of practical vocabulary, sure, but immersion is a great thing, even if it&#8217;s silly.The vocabulary can be a little silly, but immersion is good.  A newer concept for me is Gamification.  At it&#8217;s core, gamification is about turning stuff that you want to get done into a game.  Duolingo is a simple example of this.  Complete vocabulary lessons to level up and unlock new lessons.  When accomplishing tasks results in leveling up and earning rewards, the same compulsion that lead me to waste my teen years playing Everquest will now work in my favor.  That&#8217;s the idea, at least, and I manage to keep up with Duolingo fairly well.  HabitRPG is MUCH more ambitious with this concept.  It goes all out, giving you an avatar with the ability to earn mounts, group up with friends, and fight bosses.  You get to choose your &#8220;quests&#8221; so it can be tailored for whatever purpose you&#8217;d like.  If you set a daily goal to study for one hour, you&#8217;ll gain experience points and gold for completing it, but take damage each day you feel to check it off.  You can then spend accumulated gold to buy rewards that you specify, or in-game swag.  There&#8217;s no doubt that consistency and repetition are the keys to setting up positive habits.  This is where a tool like HabitRPG shines- it rewards positive habits in a fun and imaginative way.  I&#8217;m setting up my HabitRPG account to encourage more productive use of my free time.  In a future blog post, I&#8217;ll reflect on how gamification can be used in the laboratory.",
        "url": "/?p=402"
    }
    ,
    
    "p-391": {
        "title": "Lab notebook software",
            "author": "BradfordCondon",
            "category": "",
            "content": "Last year, I wrote about using Evernote as my digital lab notebook.  With the release of Findings, a new digital notebook software from the people who created my favorite reference management software Papers, I thought I would reflect on my digital notebook needs.A digital notebook should be:  Indexed and searchable, with both automatic (embedded text searching)  and manual (tags) search functions.  Evernote handles this quite well- I can easily manage what tags I have available and organize them how I wish, but searching my notebook will also find terms in embedded word documents and PDFs, for example.  OS integration.  The great thing about drinking the Apple koolaid is these apps can work well across platforms.  We don’t tend to have our laptop on our experimental benchtop; being able to pull up a notebook on my iPhone is great.  Multimedia friendly.  My notebook is a mix of text, snapshots, data annotated in Powerpoint, excel files, word documents, and PDFs.  Again, Evernote handles this quite well.  It falters a bit in flexibility when printing out my notebook- usually my images don’t come out formatted quite right and I end up with a single image per page.[]  A sample printout from my notebook. Note that I have a mix of media types (annotated powerpoint files, notebook scans, raw text). While Evernote holds them all seamlessly, printing to this PDF (and into a paper notebook) results in clunky blank pages. There is also no support for printing header information: ie experiment name, tags, dates… I have to manually write the file name on each page afterwards.  Traceable.  Ultimately, a lab notebook is for tracing the lineage of data.  Whether this is at the troubleshooting or the writeup phase, I need to understand what the starting material, protocol, and resulting output were at each stage of the experiment.  Science is seldom perfect, and a good lab notebook can prevent some confusing mixups (was that DNA sample prepared before or after I optimized the pH of buffer X?)  Here is where Evernote isn’t perfect, largely because this is a science problem.I’m looking forward to trying out Findings and reporting back how it improves on these key issues (and others I haven’t thought about).",
        "url": "/?p=391"
    }
    ,
    
    "p-389": {
        "title": "&amp;#8220;Truly extraordinary,&amp;#8221; &amp;#8220;simply not credible,&amp;#8221; &amp;#8220;suspiciously sharp:&amp;#8221; A STAP stem cell peer review report revealed",
            "author": "BradfordCondon",
            "category": "",
            "content": "Wow. These reviews are extremely incriminating; reviewers obviously spotted some of the key problems with this paper. “This paper claims that cells from any somatic tissue can be reprogrammed to a fully pluripotent state by treatment for a few days with weak acid.This is such an extraordinary claim that a very high level of proof is required to sustain it and I do not think this level has been reached”",
        "url": "/?p=389"
    }
    ,
    
    "p-378": {
        "title": "Who are CyberVor?",
            "author": "BradfordCondon",
            "category": "",
            "content": "Well, the BBC and The New York Times have both published pieces on the Russian hackers “CyberVor”.  The claim is that  1.2 billion user names and passwords from some 420,000 websites have been hacked.  The sites/users affected, nature of the vulnerability, and severity of the threat have not been disclosed.Skeptics have pointed out that, well, things don’t really add up.  The biggest problem is that The New York Times was fed the CyberVor piece by Hold Security firm.  This is the very same firm that stands to profit from this security breach, by charging $120/year for their services.  The New York Times piece, to my eye, does not validate the information provided by Hold Security.  The truth is that its in Hold Security’s interest to exaggerate the breach, and in The New York Times’ interest to report the story as quickly as possible.  Without released facts or data, this entire story could have been fabricated by Hold Security.  This is unlikely, as The New York Times piece claims that two unaffiliated sources verified the database as authentic.  Still, experts seem to think the threat could be exaggerated.So what does Hold Security know about CyberVor?  According to The New York Times piece (which means according to Hold Security, who sell the solution to the CyberVor problem), CyberVor is made up of fewer than a dozen men in South Central Russia.  Hold Security knows this because they have been in communication with them.  Seriously.I’ll be interested in seeing how this develops.  As it is, I see a lot of big claims with no evidence or specifics, and the group making the claim profiting from the resulting panic.  I also find it odd that CyberVor and Hold Security communicate.  Is that normal?  Do hackers usually chat with data security firms?  I can’t even verify that CyberVor is a thing from anyone other than Hold Security because, well, Hold Security coined the term CyberVor, and that is all we have to go on.As other skeptics have advised, however the threat plays out, taking cyber security seriously is a wise decision.  I recently invested in password management software, and highly recommend it. ",
        "url": "/?p=378"
    }
    ,
    
    "p-338": {
        "title": "MOOCs and me",
            "author": "BradfordCondon",
            "category": "",
            "content": "MOOCs (Massive Open Online Courses)  are an interesting development in higher education.  At their most impressive, they are a way for brilliant educators to reach thousands of students at a time, all across the world.  MOOCs have the potential to remove educational barriers like learning disabilities, economic constraints, geographic realities, or busy life schedules.  My alma mater, Cornell, launched their first wave of MOOCs last semester.  Cornell prides itself on catering to “any person, any study” and its not hard to see how MOOCs can drive this mission forward.Of course, MOOCs are entering the scene at a time when higher education is reconsidering its educational tenets.  A traditional classroom brings to mind stuffy tiered lecture halls with esteemed professors reciting knowledge to enraptured students (Tom Wolfe’s I Am Charlotte Simmons comes to mind).  Educators have long suspected that these teaching methods are not ideal, but recent high profile publications have provided clear evidence that there are better ways to engage students.  Active learning is a broad term, but it encompasses classrooms where students participate in activities or discussions. Instead of tuning out a professor (intentionally or not), a student must engage the material.MOOCs certainly don’t have to embody one form of education or another.  They have the capacity to be very active educational formats.  Discussion boards, problem sets, live text or video chats, course projects, and peer review all require students to work alone or together to master material.  On the other end of the spectrum, a MOOC can be a string of youtube videos or reading assignments, with multiple choice quizzes at the end of each section.  The value of such courses is questionable.  When a MOOC is offered for free, it’s not a big problem.  But as institutions offer “online certificates” for participation, the issue becomes an important one.I’ve been participating in the Data Science course track on Coursera, and am rounding the bend on the second module.  This sequence of courses is put together by Brian Caffo, Jeff Leek, and Roger Peng at John Hopkins University.  I think that theres a lot of positive examples in this course.  The community message board is quite active, with attentive TAs who field questions.  Much of the coursework is active- each week I’ve been asked to write my own code to accomplish certain tasks.  The course itself is crowdsourced for grading, a sort of peer-review lite.  The swirl() modules deserve a special commendation- these educational R packages teach you how to perform tasks in R right within the R environment.On the other hand, the weekly video lectures don’t offer very much beyond some light structure for the course.  Skipping them and relying on a search engine for the quizzes is more economical and less frustrating (nothing is worse than listening to a lecture series and taking extensive notes, only to be quizzed on what I view as minutiae).Overall I’m very grateful to the professors for putting the course together, and even more grateful that they offer it for free online (coursera offers an optional paid certificate).  Where the course succeeds, it exemplifies the potential for active learning in MOOCs.  At the end of the day, we have to remember that this is the internet.  There are seemingly limitless resources for education available.  What we demand from formal instruction is mentorship, guideposts, motivation, and accountability.  We can watch a video lecture any day- coursework should ignite our curiosity to work with the material ourselves (or at least hold a deadline over us to demand we do!).  Ultimately, we won’t learn unless we’re motivated and take the time to do so.If you’re interested in learning more about active learning practices, I recommend using the resources that “Centers for Teaching Excellence” at various universities (including Cornell) have put together.",
        "url": "/?p=338"
    }
    ,
    
    "p-281": {
        "title": "Scientific writing is dry (but that&amp;#8217;s OK)",
            "author": "BradfordCondon",
            "category": "",
            "content": "This morning I was delighted to come across the IMA Fungus journal.  Put out by the International Mycological Association , the journal title (which primarily publishes species announcements and descriptions) playfully introduces each organism with a friendly “Hello, I’m a fungus!”  The truth is, scientific literature is seldom so fun and informal.When I submitted my PhD dissertation draft to my committee, perhaps the most common complaint I got was that I was having too much fun with the writing.  Given the choice, I  always prefer to be playful than bland in my writing.  In science, this isn’t necessarily a good thing.  Primary scientific writing values clarity and correctness.  In trying to evoke vivid descriptions, I like to describe science with flair, pushing my metaphors to the absurd.  For example, I wrote that fungi forge weapons to wield against the parapets of plant defense, or that they brew novel molecules in musty cellular laboratories.The problem is that while I have a lot of fun writing statements like this, it empowers organisms with sentience and intent.  It is incorrect.  So such statements are refined to purge embellishments (fungi produce molecules that specifically target host defenses, and produce diverse secondary metabolites in specialized cellular structures).  Me being dangerously informal during my exit seminarObviously we should strive to be correct in our communication of science.  When publishing, it must be clear what we did, what we observed, and what we believe these results mean.  We must write knowing that english is not the primary language of those who are interested in our experiments.  But does this mean that we can’t have any fun?  I can’t speak for all scientists, but my love of biology is not sterile and precise.  It is colorful and messy.  It is most certainly anthropomorphized (despite my higher faculties being well aware that fungi are not sentient creatures).We scientists are a fun bunch.  This fact is more than evident from any academic conference (take the raging party that is the Fungal Genetics Conference).  But our writing is often denser than German fitness bread.  Maybe it’s impossible to get away from that in our formal writing without introducing ambiguity or compromising our professionalism.  I suppose some day I’ll find out.",
        "url": "/?p=281"
    }
    ,
    
    "p-271": {
        "title": "How many copies of my thesis/dissertation should I order?",
            "author": "BradfordCondon",
            "category": "",
            "content": " You’re finished!  You’ve spent hours writing and rewriting your thesis or dissertation (such semantics!), braved your defense, and spent even more time addressing your committee revisions (to say nothing of the years of hard work that got you here).  Congratulations.  Time to submit and watch Buffy the Vampire Slayer for eight days straight, right?  This sums up my feelings on graduation nicelyA couple of issues I hadn’t thought about came up before I clicked submit, however.  One of these was how many copies of my dissertation I should order.  I had no clue, and I was pretty braindead by this point.  So I thought I’d list some people who might want a copy for future generations of sleep-deprived and deadline flaunting scholars.  If you forget somebody, that’s OK- I was able to re-order a single copy for ~$80.FamilyYour parents, and perhaps grandparents, are a given.  After 4+ years of not being able to explain your work to anyone, having a 5 pound coffee table book to gesture at is going to make their day.  Don’t forget to include them in the acknowledgements!FriendsI didn’t order a copy for any friends, but that doesn’t mean you won’t want to.  Finish before a friend who started the same year?  Nothing rubs it in their face quite like a printed copy of your dissertation.  It sure is an expensive snub, though.Your advisorDo not forget a copy for your advisor.  Again, write something nice in the acknowledgements or pay the price.  I learned the hard way that “To my favorite advisor” isn’t acceptable.Your committeeDid you receive a lot of guidance from your committee, or shed a lot of tears in their office?  This can be a nice gesture of thanks.  I keep my dissertation on my person at all times (if only to stop bullets)Your labThis is the one place where someone might open your dissertation for very practical purposes.  Over the course of my time at Cornell I’ve consulted the dissertations of others surprisingly often, be it for a unique perspective on the literature or to see the nuances of methodology not written out in peer-reviewed publications.The libraryYour campus library probably prints a copy of your dissertation for its own purposes, but its worth checking that this isn’t expected of you.  How else could you perform the infamous $20 experiment, hiding a crisp bill in the meat of your dissertation and returning year after year to see if its disturbed?  Also, some departments keep separate dissertation collections- see if you are expected to contribute!YourselfThis probably goes without saying.  You may feel like you never want to see this cursed book again, but you probably will.  Even if its only to drag it out as a war story, or to prove to your date that you really _can _run a southern blot.",
        "url": "/?p=271"
    }
    ,
    
    "p-299": {
        "title": "Celebrity burgers a reality",
            "author": "BradfordCondon",
            "category": "",
            "content": "Back in September last year, I explored some possibilities that the “Stem Cell burger” opened up.  Perhaps the most far-fetched idea I had was to produce meat from cloned celebrity stem cells.It seems this idea wasn’t as absurd as I thought.  A new company has launched called Bitelabs, claiming they will produce celebrity-derived meats in salami-form.So is it a hoax or a real company?  The internet isn’t quite sure what to make of it.  My previous post covered the feasibility of such an endeavor, but the ethics remains a gray zone. In the meantime, I fully support the discussion, and I applaud Bitelabs’ understated strategy.  Personally I’d be interested in trying a cloned meat product, regardless of who the donor was.",
        "url": "/?p=299"
    }
    ,
    
    "p-267": {
        "title": "Scoop.it! and content curation",
            "author": "BradfordCondon",
            "category": "",
            "content": "In the past, I’ve written about the importance of setting up custom newsfeeds for current publications in your field, and using Citeulike to share references.  While these tools can be useful for discovering and archiving content, they are not ideal for sharing and disseminating it.  Gathering content and distilling it for others is an art: the art of content curation.Lately I’ve been using Scoop.it for content curation and outreach.  I created two main channels: one for science stories in the news, and another for new research in my field of fungal genetics.  By picking a key paragraph from the discussion or abstract, as well as a catchy image, I can easily pin small blurbs for each article (called scoops) to a news stream.  Each feed is easy to skim through, can be tagged and annotated, and allows me to easily review articles I was interested in.  RSS feeds for each topic can be fed elsewhere, such as a flagship brand website.  Aggregated scoops on a streamPerhaps most importantly, Scoop.it integrates other social media quite handily.  By default, I tweet all of my scoops.  Articles of general interest (generally from my science news stream) I often post to Facebook, which sometimes trigger a discussion.  This keeps me engaged in science communication, as well as in practice using social media (that I might otherwise let lie fallow).There are certainly other tools for content aggregation, but the free Scoop.it account works quite well (a subscription offers several handy features, including sharing curation duties for a topic with others).  Feel free to leave a comment recommending other services.",
        "url": "/?p=267"
    }
    ,
    
    "p-195": {
        "title": "Consider the ostracod",
            "author": "BradfordCondon",
            "category": "",
            "content": "What does it mean to be a scientist?  I present to you some overly poetic thoughts on the topic.It was a soggy morning in early Fall when I encountered an ostracod.  I had just submitted my dissertation, I was unsure about my future, I was exhausted.  To celebrate, I went hiking with a party of undergraduate and graduate students led by the intrepid Kathie Hodge foraging for mushrooms.  We had chased the rain but missed its fungal fruit, finding only a small handful of mushroom gold, Chanterelles.  We decided to return home, and on the walk back,  came across puddles housed in muddy tire tracks.  They were teeming with tiny life.  The sleek tadpoles, buzzing flies, and squirming larvae held our attention for a while, but we soon spotted a most unusual creature.Laying still, it looked like a small pebble the size of a grain of rice, its surface coated in slimy mud.  Careful observation found that the pebble scooted around the puddle in small circles, propelled as if by an onboard motor.  Enthralled, I plucked one and teased it on my finger.  What was this microscopic backwoods clam?  Amazingly, a tiny shrimp-like head poked out, with sweeping tendrils.  I immediately announced we had discovered a lost species of clam-shrimp, and found a nearby coca cola bottle to collect one.  Before we could gather another, an ATV rode by, crashing through the puddles and hopelessly clouding our view.Arriving at the lab, we armed ourselves with a microscope, and the knowledge of the scientific community.  Kathie immediately decided we were dealing with an ostracod.  An ostracod. Photo from Wikipedia: http://en.wikipedia.org/wiki/File:Ostracod.JPGObserving our ostracod (we named her Sue) scooting around under the microscope was a joy.  My curiosity for Sue was fed by an excellent website by Robin James Smith at the Lake Biwa Museum.  I highly encourage you to visit Robin’s website, especially the images section (I’ve included my favorite sketch below).  Amazingly my initial assessment wasn’t far off: the common name of my “clamshrimp” is actually “seed shrimp!”  A sketch of a female ostracod, by Robin James SmithThe amazing thing about biology is that a simple walk through the woods can ignite an entire afternoon of looking through a microscope and learning something new and bizarre.  I now know that ostracods can produce spermatozoa up to ten times their body length, can be bioluminescent, and can hunt in groups!  Every organism has an untold story, with so much left to learn and be discovered.  And thanks to the unbelievable scope and rigor of science, a discovery can easily climb the scaffolds of human knowledge.  We can just as easily reach the peak, and wonder at the reality just beyond our comprehension.I don’t think one needs a PhD to appreciate the beauty of the world.  Anyone who is excited to explore life, big or small, is a scientist in some way.  The life of a ‘real’ scientist is filled with lab notebooks, presentations, grant proposals, budget forms.  But these things are carried out in exchange for that spark of wonder that can be found in a muddy pond in the woods, in our blood, or floating in the Atlantic.",
        "url": "/?p=195"
    }
    ,
    
    "p-216": {
        "title": "The Tannosome",
            "author": "BradfordCondon",
            "category": "",
            "content": "An article posted on the Scientific American blog caught my attention recently.  Researchers have discovered the existence of a new organelle, the tannosome, in plants.  It is this organelle that produces tannins, of great interest to wine lovers.  The scientific article in question, posted to my scoop.it, was officially published in the Annals of Botany on September 25th 2013, but it barely created a ripple in the news cycle.  Why not?Organelles are extraordinarily important in biology.  An organelle is, by definition, a membrane-bound compartment within a (eukaryotic) cell that performs some specialized function.  Much like each organ within the body performs a specific task, organelles within the cell include the DNA-replicating and transcribing nucleus, the “energy factory” mitochondria, and the “storage center” vacuole.  (Most) plants possess chloroplasts, organelles which house the photosynthetic machinery required to harness energy from light.  A diagram of the organelles in a typical animal cell.Tannins are known as secondary metabolites.  This term covers a huge range of array of chemically diverse molecules, lumped together for not being “primary” metabolites, such as proteins, sugars, or ribonucleic acids (such as DNA).  Often, but not always, secondary metabolites are important for “non-primary” processes, such as herbivory defense, pheremones, or pigmentation.  Tannins can be found in most plants, and may function in herbivory defense, pathogen defense, and UV-protection.  Humans are quite fond of tannins for their role in flavoring beverages, particularly wine.The most digestible evidence they offer for the tannosome being a novel organelle is a set of TEM images (below).  On the left (A) is a standard chloroplast, and the middle and right images (B, C), show the differentiated form at different stages.  From Brillouet 2013Prior to this publication, there were reports of Tannins being found in chloroplasts- but no one had the idea that these were specialized unique organelles.  Similarly, the authors point out that chloroplasts _still produce _a number of secondary metabolites (phenolics, for example) despite this new organelle.  From what I can gather, the authors don’t make a case for this organelle producing anything other than tannins.So why the media silence?  One article notes that with a lead-in to tea and wine, it really should have caught some buzz.  Not to mention the discovery of a new organelle (albeit derived from a known organelle) is noteworthy in anyone’s book, even if it doesn’t have obvious implications past tannin biosynthesis.  For whatever reason, it just isn’t the discovery that ignites people’s interest.  Science news is generally dominated by medicine, astronomy, sex, nutrition, and the cute or gross.  Perhaps if the finding had more implications for alcohol production or consumption, the media would have taken notice.  Maybe when tannosomes are harnessed to produce a new altered-tannin  wine, the world will take notice.  But without this study, one would not even know where to begin such a task.Sources:  Brillouet JM, Romieu C, Schoefs B, Solymosi K, Cheynier V, et al. (2013) The tannosome is an organelle forming condensed tannins in the chlorophyllous organs of Tracheophyta. Ann Bot 112: 1003–1014.Organelle image: from wikimedia commons, http://upload.wikimedia.org/wikipedia/commons/thumb/1/11/Animal_Cell.svg/800px-Animal_Cell.svg.png",
        "url": "/?p=216"
    }
    ,
    
    "p-197": {
        "title": "Ethical implications of the Stem Cell Burger",
            "author": "BradfordCondon",
            "category": "",
            "content": "I am  writing a short piece on the Stem Cell Burger which was unveiled early last month.  The Cultured Beef website is eager to point out the many benefits this technology could have for global research usage and animal ethics.  Opponents are just as quick to point out the possibility of unknown side-effects.  Cultured Beef may change the world- but maybe not in ways people expect.  I&#8217;ve jotted down a few of the more extreme implications for Cultured Beef.  I&#8217;m not an actual proponent of all these ideas: this is really just a thought experiment for possible ethical issues downstream of culturing just beef.1)  Eating endangered species.Why limit ourselves to cows?  If we can string together cow cells grown in culture, why not the cells of a Bald Eagle?  It would be arguably the most American burger possible, and wouldn’t require death or suffering from of our favorite bird of prey.  Bald Eagle Burger is a real band! The burgers aren’t real. Yet2)  Eating extinct species.With scientists pondering the implications of cloning extinct species, such as the woolly mammoth, why not clone these extinct species into the Cultured Beef pipeline?  I imagine lots of people would be very excited to eat woolly mammoth burgers.  Mammoth ribs, the original family meal3)  Eating people.Cannibalism is one of those taboos that people love to joke about, but when push comes to shove, few in our modern society would seriously consider.  But what if there was a meat you could buy that came from people, _without anyone being hurt or killed?  _And we aren’t just limited to eating any old person.  Cultured cells of celebrities, dead and living.  Bill Clinton hot dogs.  Miley Cyrus patties.  American celebrity culture is ready.  Forget energy drinks, stem cell burgers could be the next big endorsement.4)  Eating ourselves.Why stop at eating other people?  We could custom-sell Cultured Beef to be made from an individual customer’s cells.  Depending on the source of broth nutrients, this could be an entirely new movement in veganism.  No more annoying jokes about murdering vegetables.  I’m 100% self-supporting, subsisting on only genuine Me-burgers grown in bacterial broth.Note that I don’t condone cannibalism, or Miley Cyrus, and I think that Cultured Beef is a great idea.  What about you?  If you could culture an organism’s cells and eat them, what would you pick and why?",
        "url": "/?p=197"
    }
    ,
    
    "p-188": {
        "title": "Twitter bootcamp",
            "author": "BradfordCondon",
            "category": "",
            "content": "Twitter bootcamp starts today!  It’s put together by my fellow Oberlin alumn Ma’ayan Plaut, who is a social media role model.  If you’ve never quite “gotten” Twitter, or if you feel like you aren’t getting enough out of it, check it out!  I’ll be participating!",
        "url": "/?p=188"
    }
    ,
    
    "p-86": {
        "title": "RSS readers",
            "author": "BradfordCondon",
            "category": "",
            "content": "(This post is a continuation of my science RSS series, found here)Once you’ve identified RSS feeds you’d like to follow, you need to choose an RSS reader.  This is where the fruits of selective RSS subscription are much sweeter than signing up for email notifications.  Notifications go to my inbox, where they are drowned out by the day-to-day requirements of the academic life.RSS feeds, on the other hand, wait patiently in the reader, separate, sorted, and ready to be skimmed.  Flipping through my reader every few days, and posting  articles of interest to citeulike, or Scoop.it (which I haven’t written about yet, but will), I stay on top of the big news in the science world, and the details relevant to my research.  Your reader may sync up to the cloud, your social media, or other reading, storage, or notetaking apps.  You can use your RSS reader to skim headlines and send the interesting bits to other programs, or you can do all your reading in the reader, depending on your personal preference.But what reader to use?Google Reader used to be the reader of choice.  Unfortunately, Google is discontinuing reader, and several readers have risen to fill its shoes.I currently use Feedly.  It syncs with my Google account, and lives in my browser, in harmony with my work and home computer.  I have my feeds roughly divided into peer reviewed publication, science news, and play/misc., which is good enough for me.  Feedly sports an ipad app, which is slick and intuitive.  Oh, and it’s free, which is a huge plus.  The feedly ipad interface, standard issue. I removed all the default categories.Cnet has reviews of other RSS readers. They come in all shapes in sizes, from feature heavy standalone apps to minimalist browser integrated readers.  Shop around and go with what feels good- the most important thing is that you can integrate it into your work flow.  I used to use RSSowl for my feeds, which is also free.  But, as a standalone app, I used it less and less, until I stopped checking my feeds all together.  The most important rule for any app is that you use it!For me, I will use a program if it fits well into my existing workflow.  I skim headlines in my reader.  Items of interest are sent to either pocket, evernote, citeulike, or scoop.it, depending on if they are for my personal consumption, work-related, or science outreach and communication related.  I will expand on these specific avenues in a later post.",
        "url": "/?p=86"
    }
    ,
    
    "tools-2013-07-18-a-digital-notebook": {
        "title": "A digital notebook",
            "author": "BradfordCondon",
            "category": "",
            "content": "When I first started graduate school, I kept an old-fashioned notebook.  A 4×4 quad ruled Roaring Spring Lab Notebook.It had gridlines, it had carbon paper (The carbon sheets remain, unperturbed, at the back of the notebook), it had Thomas Edison on the cover scienceing it up.  It was everything I thought science would and should be.As a tool for keeping track of what I did for later reference, however,it was a complete disaster.I’m not an organized person by nature.  I work in spurts, I don’t keep careful track of what I’m doing.  I often have to go back and fill in the details.  This means writeups I would intend to return to later were buried under experiments I would go ahead with in the meantime.I also am not a linear person (or perhaps science is not a linear endeavor).  One day I would work on project X, the next, project Y.  Science works that way, sometimes, project X needs a week to grow, sometimes project Y sits in the fridge while the primers are synthesized and delivered in the mail.Both of these traits make for a messy, and extremely confusing lab notebook.  Sure, I could have kept one notebook for each project I was working on.  But what about when I’d perform an experiment that fit into two projects?  Sometimes I will perform an assay using two sets of mutants, for two separate reasons.  Would I double-copy the experiment?  Perhaps this is when I should have actually used that carbon paper.No, instead, I switched to a standard 3-ring binder after my first year in graduate school.  I could re-define the boundaries of a given project easily, add and remove page dividers for each project, and easily add in printed pages.Looking back on my research project (and looking forward to where I may be next), I think if I could do it again, I would keep a digital notebook first and foremost.  I did keep a digital notebook, using the app Mori.  Unfortunately, development on Mori ceased long ago, and I used it exclusively for digital notes.No, I think the ideal thing to do would be to keep a purely digital notebook, with printed backups of results and summaries in a three-ring notebook.A digital notebook is searchable.  It doesn’t have to be linear.  I can look for all projects with a given gene or technique, and pull them all out.  A good notebook is tagable (and hopefully well tagged.)  I don’t have to decide if an experiment goes in notebook A or notebook B.  It just gets tagged AB, and goes into the digital notebook.I also don’t have to wait until my results are processed to add them to the notebook.  This is a big one.  Raw unprocessed data files (such as .nexml tree files that are a finished product, but need to be visualized, or microscopy experiments that take a fair amount of post processing effort) go right in the notebook.  A sloppily generated note on media generation gets filed away digitally.Scanning in the above sheet, for example, lets me find it no matter what project I’m working on in the future, rather than digging through years of failed and succesful experiments in different notebooks when writing my materials and methods later.  Note that I added a minimal number of tags, but wrote out the key media ingredients (different forms of iron) for search purposes.I’ll talk more about the best way to organize a digital notebook and tips for making the most of it (for example, sharing and compiling status updates for the boss).",
        "url": "/tools/2013/07/18/a-digital-notebook/"
    }
    ,
    
    "uncategorized-2013-07-14-emotion-matters-zimmerman-vs-alexander": {
        "title": "Emotion matters: Zimmerman vs Alexander",
            "author": "BradfordCondon",
            "category": "",
            "content": "The Florida Stand Your Ground controversy has everyone talking (and nevermind that neither case officially invoked said law).  And a mother (Marissa Alexander) getting 20 years (a minimum sentence for firing a gun during a crime) for firing warning shots at an abusive husband looks pretty shitty next time a man (George Zimmerman) shooting a 17-year old boy and being acquitted of all charges.What I find amazing here is the central role emotions play.  Not just in the public’s reaction to these verdicts in sentences: but in how those verdicts were reached.Both verdicts, for Zimmerman and Alexander, came down to one thing: was the shooter angry, or fearful?  The jury found that Zimmerman was afraid when he shot and killed Treyvon Martin.  Marissa Alexander, on the other hand, was found to be angry.  Angry at her husband for beating her, but nonetheless, angry.  Because she was angry, she could not utilize Stand Your Ground, and was found guilty of assault (I encourage you to actually read the reports of both trials if you are interested in either case, as in both situations a jury reached that verdict somehow.  In the case of Alexander, the prosecution made a big deal over the “warning shots” fired, and the possibility that these shots could have hit one of her children.)  The state of a defendant’s mind during a crime can matter more for the verdict than the direct consesquences of the crime itself.How can such a distinction between fear and anger make all the difference?  Is the line between fear and anger so clearcut that we can make this call, with such grave consequences?  Was Alexander not angry because of a long history of fear and abuse?  Was Zimmerman’s anger what prompted him to confront Martin in the first place?I encourage all of us to rally for justice.  Justice, we are told, is a matter for the courts to decide, not a realm of emotion.  But what happens when justice hinges on the weighing of emotion?",
        "url": "/uncategorized/2013/07/14/emotion-matters-zimmerman-vs-alexander/"
    }
    ,
    
    "p-140": {
        "title": "Papers as a reference manager",
            "author": "BradfordCondon",
            "category": "",
            "content": "I love Mekentosj’s Papers, and encourage you to check it out if you are on a mac and keep digital literature.One of the things that drove me nuts using my advisor’s Endnote library was that I couldn’t change it.  Each publication, each grant, it would be my task to go through and manually “spruce up” the references.  Each time I would plea to change the actual source library, and each time I’d be told we had no time as it was the 11th hour.  Everything was inevitably inconsistent.  Author names, journal abbreviations, italicizing species name.  What a mess.Now that I am in charge and doing my own references, I can keep a clean house.  This means fixing references the “right” way, in Papers.A good way to fix up a lot of metadata easily is to manage your periodical collection. To do this, click on your periodicals tab under sources, and sort them by name.  Merge journals that should be the same- Set a standardized way of abbreviating (I do without periods), stick with it.  A thorough periodicals cleaning will make formatting your final bibliography much easier.Are there repeats?  Merge them!  Is the abbreviated title appearing in the full title field?  Fix it!  Do it here, and you are fixing each and every affiliated journal!  Format BMC Microbiology right here, and every time an article from BMC Micro comes in, it’ll be properly formatted (provided it’s correctly linked to this journal entry!)  BMC Biotechnology has the abbreviation in both the full and abbreviated field. An easy fix!Sometimes you’ll see weird journal titles.  Are they really journals, or is it a book or chapter set to the wrong source type?  Make sure each article is set to the correct type.  A PhD chapter that thinks it is a journal article is going to be formatted incorrectly.  Enable editing to change the source type for an article.As I continue battling my own citations, I’ll post again with more tips for tidying up your references.",
        "url": "/?p=140"
    }
    ,
    
    "p-134": {
        "title": "Mycozombies of The Last of Us",
            "author": "BradfordCondon",
            "category": "",
            "content": "Warning:  probably some incredibly mild spoilers below regarding the game.The Last of Us is a new survival action horror videogame for the Playstation 3.   The game is inspired by a real life fungus, Ophiocordyceps unilateralis_.  _Any time fungi get love in popular culture, I’m pretty happy.  I’ve long thought that fungi are a hugely underused resource for science fiction, and I think that cordyceps species are starting to gain some traction in the popular press.I’ve only played a couple of hours of the game (as I’m supposed to be writing my dissertation, and don’t own a playstation).  That said, I thought I’d comment on the mycology as I come across it.  Beautiful fanart of a Clicker by thomaswievegg. http://thomaswievegg.deviantart.com/art/the-Last-of-Us-Clicker-3552776291) Clickers.These zombies have mutated to the point that their heads have sprouted fungal fruiting structures.  The problem is that their heads really strike me as bracket fungi (conks), and their shoulders are fruiting gilled mushrooms.  Ophiocordyceps are ascomycetes.  Gilled mushrooms, and conks, are produced by basidiomycetes: a phylum level distinction.  For context, two animal phyla include chordata and annelida: the distance between segmented worms and animals with a spinal chord (ie, us).  That said, Ophiocordyceps unilateralis fruits out of the head of its victim- much like a Clicker.  The Turkey Tail bracket fungus. Photo credit: wikipedia2)  The introduction videoAs the credits roll, we are treated to some weird, pulsing “fungal” growth, followed by fruiting body formation and spore dispersal.It’s beautiful, it’s creepy, it’s everything we love about fungi.  But what are we really looking at?  To me, the first bit looks distinctly like a slime mold.  This is a claim I’ve made before, in reference to_ The Blob.  _  Hyphal-like growth but the pulsing is really more like a slime mold.Next, the fungus forms fruiting bodies.  Fruiting bodies form and disperse spores.These really could pass for a cordyceps fruiting structure.  Generally, the stalks typical of cordyceps form out of the host species, which may be the case here.  In the game, areas thick with spores are encountered around freshly killed infected humans- a pleasing consistency.Watch the intro video here.Thank you Naughty Dog, for a great zombie game, and some fun biology!  Maybe if I get the chance to play through more of it, I’ll be able to write more about the fungal biology underlying the zombies (mycozombies?).",
        "url": "/?p=134"
    }
    ,
    
    "p-99": {
        "title": "Snack Writing",
            "author": "BradfordCondon",
            "category": "",
            "content": "I was reading this Nature article by Kim Gardiner and Hugh Kearns on snack writing.  Their argument is that scientists who write in a small, fixed window, 45 minutes a day, every day, write more than “block writers” who blackout entire days for writing.I’ve been a block writer my entire graduate career, that’s for sure.  But I can see the merit of snack writing, especially when it comes to methodology.  The worst feeling of writing a paper is digging through stacks of ancient data from experiments you don’t remember doing, trying to remember which of the six primer pairs you designed actually worked (let alone what annealing temperature they worked at.)  Writing up results as they are generated is a Godsend, especially when things don’t work out.I have a tendency to bury things that don’t work: and why shouldn’t I?  Inconsistent, unclear, unexpected: we scientists are storytellers, and “bad” data that doesn’t fit the tale makes us feel uncomfortable.  We don’t know what it means- is our methodology flawed, or is there a fundamental misunderstanding in our thinking?  So instead, we are encouraged to focus on experiments that do work, and build our story from there.But you never know when you will double back, and find yourself facing the ghosts of unpleasant experiments.  Even if you keep a meticulous notebook, data without the narrative can be confusing.  So spare yourself the trouble, and set aside a time when you generally aren’t very productive, and regularly wrap up what you’ve done and why you did it.",
        "url": "/?p=99"
    }
    ,
    
    "p-111": {
        "title": "Scientifink!",
            "author": "BradfordCondon",
            "category": "",
            "content": "I am featured in this week’s CALS “scientifink” writeup!I happen to love my ascospore tattoo, and am happy to share it with the world.  ascospores on my armI remember when I got it done, after passing my A exam.  It was my first tattoo, and I felt ready to make a commitment.  After all, I was going to spend 5 (hah!) years working with this fungus: how could I possibly get sick of it?  Plus, once I passed my B exam and earned my PhD, I would build onto it with the asexual spores.  Eventually I’ll have the arm covered with organisms I’ve worked with as my career goes on.I’ve actually scheduled my B exam for NEXT MONTH!  This means that I get to add to it soon.  I’m planning on getting several more asci, as well as an emptied ascus with ascospores loosely stuck together, like in the photo also included in the article.(An ascus is a sac.  Ascomycetes are fungi which produce their (asco)spores in sacs.  This is in contrast to basidiomycetes, which produces their spores on basidium.  This is what the mushrooms you would find in the grocery store do).  ascospores under the scopeIt’s hard for me to overstate my kinship with Cochliobolus heterostrophus.  I’ve spent 6 years working with it, and love it to death.  The hours I’ve spent transferring cultures of it, spraying it on corn, collecting spores, and dissecting ascospores under the microscope have really transformed.  Now that I’m at the end of my PhD, I feel like a “real” scientist, capable of breaking down problems, with my eyes and ears open for unexpected discoveries.Tattoo photos courtesy Kent Loefflerhttp://cornellcals.tumblr.com/post/52869381183/scientifink-bradford-condon",
        "url": "/?p=111"
    }
    ,
    
    "p-104": {
        "title": "Having trouble learning?  Get some rest",
            "author": "BradfordCondon",
            "category": "",
            "content": "Do you schedule breaks and rests during your working, writing, and learning?  Do you make time for sleep in your life?   A new article by Ingrid L. C. Nieuwenhuis et al., published June 5th in the journal PLOS ONE, provides another piece of evidence that you should.  Sleep is an important step for proper learning. The details of the study involved exposing participants to complex strings of letters that followed unknown rules, and then teaching the participants the rules.  They would then leave the lab and come back 15 minutes, 12 hours (sleeping), or 12 hours (awake) later to retest.  The authors found that the group who slept  performed better at classifying the letter strings upon their return- sleep played some role in understanding the grammatical rules they were taught.This is one piece of information in a body of literature that suggests we make information network connections in our sleep.  Learning can be thought of as a web- every fact is connected to another, and another.  Understanding how things are connected is a powerful way to learn.  For example, in Biology, we often classify organisms according to Kingdom of life.  We then learn what attributes plants and animals have in common, and what traits separate them.   This study shows that sleep may be an important part of processing, and connecting, such information. Here at Cornell, the all-nighter is an all too common occurrence.  This may be fine for simply retaining facts, but if your exam challenges you to make connections between your information, or demonstrate higher learning, you might be better off getting a good night’s rest.   Article source:Nieuwenhuis ILC, Folia V, Forkstam C, Jensen O, Petersson KM (2013) Sleep Promotes the Extraction of Grammatical Rules. PLoS ONE 8(6): e65046. doi:10.1371/journal.pone.0065046  Photo source:http://en.wikipedia.org/wiki/File:Sleeping-girl.jpg",
        "url": "/?p=104"
    }
    ,
    
    "p-58": {
        "title": "Multitasking help: Slicer",
            "author": "BradfordCondon",
            "category": "",
            "content": "Sometimes it’s the little things that make a big difference.Slicer is a tiny app.  Its sole function is to reduce the active screen window to 50%, 33%, or 66% of the screen height.What do I use Slicer for?  When I’m copying data from a figure or excel file into a text writeup.  When I’m browsing the internet and taking notes at the same time.    Reorganizing data folder structures, dragging from one window to another. Anytime I’m multitasking.   Or half-tasking (procrastinating and working at the same).  And when am I not doing one of those two things?  Slicer makes hopping between apps, and working with multiple apps at the same time, so much easier.This blog is really about ways to be a better scientist; efficient use of technology is one of them.  Slicer is one of those apps that I can’t believe I rely on so heavily- was it  so bad to drag and resize manually?  All I can say, working in a post-Slicer environment, where all of my windows fit into one of my designated work zones perfectly, I could never go back to my primitive ways.",
        "url": "/?p=58"
    }
    ,
    
    "p-60": {
        "title": "Four ways to get writing",
            "author": "BradfordCondon",
            "category": "",
            "content": "I am trying to write my PhD dissertation.It’s going OK, thanks for asking.  But it’s not going OK all the time.Some days, I can spend an hour in front of my computer and have nothing to show for it but a lot of purple links on Reddit.  Surfing the web instead of working? Sounds like time for a run to me.Other days I can literally write, almost nonstop and without distraction, for epic 10 hour sessions.  So what’s the trick?  Here is my experience.#1.  Deadlines.If you are writing your dissertation, then it’s a safe bet you can meet a deadline.  Be it a term paper, grant application, or conference abstract, my life is littered with deadlines.  And, the week before a deadline, two things happen.  I put down other interesting projects, and I write productively because it has to get done.But the problem is that a dissertation is such a huge task that it inhabits its own epic time frame.  It will be finished when it is finished, it won’t be shoved out in a week or a month.  But setting reasonable, artificial deadlines for small pieces of my thesis has allowed me to step out of the lab, enjoy some of that deadline-induced focus, and put digital ink to screen.#2.  Write when you can write.Why fight it?  I can’t always write.  If it’s a nice day out and I can’t concentrate, I go out and play. If I can’t write because I’m thinking about my sister, I call her.  If I’m hungry, I eat.By the same token, sometimes I can write.  On these days, I cancel commitments and appointments.  I live that day as if writing were the only task on my desk.Rather than seeking to control my productivity, I work with it.#3.  Write the shitty, plagiarized first draft.I am not  a perfectionist, really.  I’m not even a details person.  But when it comes to writing, it’s really tough for me to get going.  Oddly, the toughest section is writing the introduction (more on that later, perhaps) where what I need to say has already been written, a thousand times.  It’s perfect, because I didn’t write it! And for right now, that’s OK!By blatantly plagiarizing whole sections from other people’s reviews and my own writing elsewhere (and highlighting the sections so I don’t forget, of course), I can move on, revise, and build off of what’s on the page.  I no longer try to rewrite what has been said again and again (a more unpleasant task, to me, than blazing a new trail with new data)- I go with what’s been said, and after things are in place, I reshape it to my needs.#4.  Get active.Seriously.  Go to the gym, go for a run, shoot some hoops.  Sometimes I just clean my house for an hour (in fact, I think my apartment is only clean when I have a deadline looming.  This may not be for everyone, as it’s a trait I share with my mother, but not my father.).  I leave the music at home, and busy my body while clearing my mind.  When I return to my desk, I’m often reset and ready to work.If you are reading this, then I wish you the best of luck on your project.  Get back to work!  And if you can’t, and you don’t have a real world deadline breathing down your neck, go take care of yourself!  Because you deserve it, and you’ll function better if you aren’t burnt out.",
        "url": "/?p=60"
    }
    ,
    
    "p-51": {
        "title": "Get your work on the cloud: Dropbox",
            "author": "BradfordCondon",
            "category": "",
            "content": "Dropbox is simple and intuitive.  Make a folder on your computer that is automatically backed up to a cloud storage account.  Suddenly, the files you work on and the data you access syncs across devices, at home and at work.  Plus, you can dropbox folders with others.Write at home, go in to work, and the latest version is on dropbox without passing it between USB sticks.Collaborating on a manuscript?  Keep the current manuscript and reference library in a shared folder.  No more emailing files back and forth (and, most frustratingly, losing comments or edits as versions are passed back and forth.  In theory, Microsoft Skydrive should be the ideal solution for this sort of thing, as it boasts seamless editing of word files as google drive does.  Unfortunately for both of these services, I wasn’t able to get it to seamlessly integrate as well without doing the word processing on the web interface.  Fun fact: if it doesn’t work easily for the boss, it isn’t going to happen.Fortunately, collaborators tend to be elsewhere in the world, and while I’m a night creature, my advisor writes in the morning.  This means we aren’t all trying to edit the same Dropbox shared document at once.Another nice use for papers is syncing my Papers library.  I wish that Papers had this kind of service automatically, but it’s a nice work around.  Keep your papers library on dropbox in a secure location, and you can sync it between macs.  Just be careful to close down Papers when you are done with it – otherwise, you’ll get conflicting libraries.There are a million other uses for Dropbox, but these are the ones I regularly take advantage of.",
        "url": "/?p=51"
    }
    ,
    
    "p-42": {
        "title": "Let the literature come to you: custom pubmed RSS feeds",
            "author": "BradfordCondon",
            "category": "",
            "content": "RSS (rich site summary, or, really simple syndication) is an invaluable tool for all academics.  If you ever wish it was easier to stay up the current academic literature, then this is tool for you.So what is RSS?  It’s a way of publishing work so that readers can be notified of updates.  In this day and age, almost every journal has an RSS feed- all you have to do is hook your RSS reader up to the journal’s RSS feed, and you will have every new issue delivered to your virtual doorstep.  You’ll find RSS feeds everywhere: there’s one lurking at the bottom of the sidebar.The list of journals I read cover to cover is very low (well, it’s nonexistent).  The utility of RSS starts to become apparent when you combine it with search terms.  Pubmed makes it insanely easy to do this.  First, make a literature search with some well-devised keywords and observe the results.  Pick terms that are specific enough to filter out things you have no interest in, but are not so demanding you find no results.  Create a custom RSS feed with the right search terms and you’ll never have to skim the table of contents for a journal againOnce you are satisfied with your search terms,  click on the link to save your search as an RSS.  Now, whenever a new article matching your search term is published and collected by pubmed, it will be pushed to your RSS feed.  Combine several of these RSS searches, and you will impress your advisor/journal club/dog with your awareness of cutting-edge research in your field!Next time: picking an RSS reader!",
        "url": "/?p=42"
    }
    ,
    
    "p-6": {
        "title": "Citation management: Papers",
            "author": "BradfordCondon",
            "category": "",
            "content": "Before you can do good science, you must read good science.   A lot of it.  The good, the bad.But how to manage your scientific reading?  Do you print out everything you can possibly hope to read some day?  File it away somewhere?Maybe you download PDFs, and carefully curate your articles in a dense folder structure forest.Being the supremely lazy and scattered-brained person that I am, I need a more hands-off solution.  The primary tool I use for reading and storing scientific papers is Papers.  Think of it as itunes for your PDF library.   Download pdfs, drag them into papers, and let it do the rest.  The program is quite good at automatically pulling meta data (author, journal, etc) from the web.  I point mine to pubmed only: google scholar metadata isn’t always spot on.Papers can be read, rated, and tagged.  Best of all, there is ipad integration with automatic library syncinc and highlighting.  This means that I effectively never have to print out a paper again.  I bring my ipad to discussions or as a side-car to my workstation when writing papers.My qualms with papers really relate to my own inadequacies.  For some reason, collections have never been quite intuitive enough for me, so my papers are a hugely disorganized mess.  Tagging papers with keywords, and then creating smart collections around those keywords, sounds like a good strategy, but not one I’ve implemented.  Fortunately, the spotlight index is really robust, and I can find what I’m looking for without too much fuss. I’ve also heard good things about Mendeley and Xotero- feel free to leave a comment about which you prefer.",
        "url": "/?p=6"
    }
    ,
    
    "p-16": {
        "title": "Tools for science: Citeulike",
            "author": "BradfordCondon",
            "category": "",
            "content": "I’m pretty sure I have too many gadgets for managing my academic references.  I’d surely be better off if I spent more time reading the literature, and less time setting up new accounts that I’ll neglect (this blog being a perfect case in point).Every year for my journal club, I try to get people to use Citeulike.  It’s a great concept- install a browser bookmark, and with one click, import all of the metadata for a paper to your personal library.  Tag it (which I always neglect), and there you go.  It’s pretty tough to stump the import tool (this wasn’t always so), so without much effort, it pulls the title, author, abstract, and journal info.  A citeulike entryWhich is all nice, but it doesn’t stop there.  Citeulike has all sorts of functionality for sharing papers with a dedicate group, or just e-stalking a scientist you have a crush on (or who stays on top of the literature).  In theory, this means you’ll be plugged into your academic community, noticing new publications and seeing who is reading what.In practice, it’s been a bit more lonely.  As it is, I use it as a citation repository for articles I haven’t read, or read but don’t want to keep in my papers library (an upcoming post on that!).  My Citeulike library is here, if you have any interest in fungal genetics, follow me!",
        "url": "/?p=16"
    }
    
    
    };
</script>
<script src="/js/lunr.min.js"></script>
<script src="/js/search.js"></script>