---
id: 484
title: CV and contact
date: 2015-12-05T03:02:17+00:00
author: bradford.condon@gmail.com
layout: page
permalink: cv.html
---


<p style="text-align: center;">
  Bradford Condon, PhD
</p>

## Education

August 2013, PhD in Plant Pathology & Plant-Microbe Interactions, Cornell University, Ithaca, NY

May 2007, BA in Biology and Religion, Oberlin College, Oberlin, OH


## Employment


February 2017 to present: Postdoctoral associate, Department of Entomology and Plant Pathology, University of Tennessee, Knoxville.

January 2016 to present: Postdoctoral associate, Department of Plant Pathology, University of Kentucky.

June 2014 to December 2015: Postdoctoral associate, Department of Ecology, Evolution, and Behavior, University of Minnesota.

August 2007 to June 2014: Graduate Student and Post-Doctoral fellow, Department of Plant Pathology & Plant-Microbe Interactions, Cornell University, advised by Dr. [B Gillian Turgeon](http://www.plantpath.cornell.edu/labs/turgeon/index.html)

<p style="text-align: center;">
  <a href="https://i1.wp.com/www.bradfordcondon.com/wp-content/uploads/2015/12/TurgeonLab_4_10.jpg"><img class="alignnone size-medium wp-image-485" src="https://i2.wp.com/www.bradfordcondon.com/wp-content/uploads/2015/12/TurgeonLab_4_10-300x166.jpg?fit=300%2C166" alt="TurgeonLab_4_10" srcset="https://i1.wp.com/www.bradfordcondon.com/wp-content/uploads/2015/12/TurgeonLab_4_10.jpg?resize=300%2C166 300w, https://i1.wp.com/www.bradfordcondon.com/wp-content/uploads/2015/12/TurgeonLab_4_10.jpg?w=510 510w" sizes="(max-width: 300px) 100vw, 300px" data-recalc-dims="1" /></a>
</p>


## Skills

#### Bioinformatics and programming

  * Proficient with UNIX/LINUX systems, command-line programming, querying and managing big data/databases (SQL, JSON)
  * Developer for cross-platform apps in React-Native, and genomics websites using Tripal and Drupal.
  * Experience designing experiments and analyzing data for a wide variety of bioinformatic projects, including 
      * Metagenomics
      * RNA-seq and transcriptomics
      * Genome assembly and annotation
      * Phylogenetics

  * Programming languages: **R, Perl, swift, javascript, PHP, CSS, Ruby**
  * Frameworks: **Laravel, Drupal, Wordpress, React Native**

#### Molecular biology

  * Fungal and bacterial culture, transformation, and phenotyping
  * Molecular bio bench skills include DNA extraction, PCR, southern/northern blotting, light and confocal microscopy
  * Next-gen (illumina) library preparation
  * Greenhouse management

#### Communication and mentorship

  * Formed UK-BLAST, the undergraduate bioinformatics mentorship group at the University of Kentucky 
      * Leading and designing workshops in bioinformatics
      * Co-mentoring and advising students on their bioinformatics projects
  * Strong track record of supervising undergraduates in the laboratory and integrating their work into complex workflows
  * Co-PI awardee for bringing [Expanding Your Horizons](http://www.eyhn.org/), a STEM conference for young girls, to the University of Kentucky
  * Community outreach and education on fungi includes presentations at Floracliff Nature Sanctuary and the Kentucky Food Retail Safety conference

#### Awards and honors

* 2006 Oberlin College Norman H Wright Prize for undergraduate research
* 2007-2008 Presidential Life Science Fellow
* 2009 NSF graduate research fellowship honorable mention
* 2010 NSF EAPSI fellow (New Zealand)
* 2011 Plant Sciences outstanding TA award
* 2012 McClintock Outstanding Plant Sciences Graduate Student Award
* 2011-2012 Center For Teaching Excellence Graduate Teaching Associate Fellow
* 2012-2013 Center For Teaching Excellence Graduate Teaching Associate Fellow
* 2017 NSF EPSCoR grant, [Expanding Your Horizons](http://www.eyhn.org/)
* 2017 [UKY SOPS](http://sops.as.uky.edu/?q=content/symposia) research symposium postdoc poster award (second place)

#### Teaching and leadership

##### KBRIN Next Generation Sequencing and Data Analysis Workshop, *University of Kentucky* 2017
 
Instructor and web developer.  Implemented and maintained the (Drupal-based workshop website)[https://ngs.csr.uky.edu/] and REDCap registration system.

##### UK-BLAST: *University of Kentucky* 2016

UK-BLAST is an extracurricular support group for undergraduates performing genomics research at the University of Kentucky.  In addition to advising students on their projects, I designed and implemented workshops on topics in genomics.


##### Departmental teaching: _Cornell University_ 2008-2013

* Spring 2008: PLPA 2010 TA: Magical Mushrooms & Mischievous Molds.
* Spring 2010: PLPA 2015 TA: Mushrooms, Molds and Molecules.
* Spring 2011 PLPA 2015 TA: Mushrooms, Molds, and Molecules.
* Spring 2012 PLPA 2015 TA:  Mushrooms, Molds, and Molecules.
* Spring 2013 PLPA 2015 TA:  Mushrooms, Molds, and Molecules.

_ _

##### [Center for Teaching Excellence](http://www.cte.cornell.edu/): _Cornell University_ 2011-2013

_workshops (1 hour) developed & implemented:_

>* Classical teaching models and the psychology of learning
* What do I want my students to learn? Identifying learning outcomes as the basis for instruction and assessment
* Using library resources to enhance student’s research skills
* Creating effective questions for iClickers and online quizzes
* Establishing a professional electronic online identity

_Full day workshops developed & implemented:_ 

>* 2012 Chemistry TA training instructor
* 2012 Teaching Excellence Practicum Instructor, CTE summer Institute

#### Presentations

##### Workshops
* [Floracliff mushroom identification workshop](http://www.bradfordcondon.com/2017/04/17/floraCliffUpdate/) Lexington, KY (2016)

##### Talks

* Joint Genome Institute Dothideomycete jamboree. Walnut Creek, CA (2008)
* Fungal Genetics Conference, secondary metabolism session. Asilomar, CA (2013)
* Kentucky Food Retail Safety Conference, invited speaker on wild mushroom safety. Frankfort, KY (2016)
  - [Slides](https://www.slideshare.net/BradfordCondon1/ky-retail-foodseminar)


##### Posters

* American Society of Plant Biologists, Boston, MA (2006)
* Fungal Genetics Conference, Asilomar, CA (2009)
* Molecular Plant Microbe Biology, Quebec City, QC (2009)
* Fungal Genetics Conference, Asilomar, CA (2011)
* [University of Kentucky Society of Postdoctoral Scholars Research Symposium](http://sops.as.uky.edu/), Lexington, KY (2017) 



## Publications

### Pre-print

Gladieux, P., **Condon, B. J.,** Ravel, S., Soanes, D., Nunes Maciel, J. L., Nhani, A., Jr., Terauchi, R., Lebrun, M.-H., Tharreau, D., Mitchell, T., Pedley, K. F., Valent, B., Talbot, N., Farman, M., and Fournier, E. (2017). Gene flow between divergent cereal- and grass-specific lineages of the rice blast fungus _Magnaporthe oryzae_. bioRxiv. :161513

### Peer-reviewed

Mohd Zainudin, N. A. I., **Condon, B. J.,** De Bruyne, L., Poucke, C. V., Bi, Q., Li, W., et al. (2015). Virulence, host selective toxin production, and development of three Cochliobolus phytopathogens lacking the Sfp-type 4&#8242;-phosphopantetheinyl transferase Ppt1. _MPMI_ (digital preprint). doi:10.1094/MPMI-03-15-0068-R

Santana, M. F., Silva, J. C. F., Mizubuti, E. S. G., Araújo, E. F., **Condon, B. J.,** Turgeon, B. G., & Queiroz, M. V. (2014). Characterization and potential evolutionary impact of transposable elements in the genome of _Cochliobolus_ _heterostrophus_. _BMC Genomics_, _15_(1), 536. doi:10.1186/1471-2164-15-536

**Condon, B. J.,** Wu, D., Kraševec, N., Horwitz, B. A., & Turgeon, B. G. (2014). Comparative Genomics of _Cochliobolus_ Phytopathogens. In R. A. Dean, A. Lichens-Park, & C. Kole, _Genomics of Plant-Associated Fungi: Monocot Pathogens_ (pp. 41–67). Berlin, Heidelberg: Springer Berlin Heidelberg. doi:10.1007/978-3-662-44053-7_2

**Condon, B. J.,** Oide, S., Gibson, D. M., Krasnoff, S. B., & Turgeon, B. G. (2014). Reductive iron assimilation and intracellular siderophores assist extracellular siderophore-driven iron homeostasis and virulence. _Molecular Plant-Microbe Interactions : MPMI_, _27_(8), 793–808. doi:10.1094/MPMI-11-13-0328-R

Horwitz, B. A., **Condon, B.**, & Turgeon, B. G. (2013). Cochliobolus heterostrophus: A Dothideomycete Pathogen of Maize. In _Genomics of Soil-and Plant-associated Fungi_ (Vol. 36, pp. 213–228). Berlin, Heidelberg: Springer Berlin Heidelberg. doi:10.1007/978-3-642-39339-6_9

Zhang, N., MohdZainudin, N. A. I., Scher, K., **Condon, B. J.,** Horwitz, B. A., & Turgeon, B. G. (2013). Iron, oxidative stress, and virulence: roles of iron-sensitive transcription factor Sre1 and the redox sensor ChAp1 in the maize pathogen _Cochliobolus heterostrophus_. _Molecular Plant-Microbe Interactions : MPMI_, _26_(12), 1473–1485. doi:10.1094/MPMI-02-13-0055-R

**Condon, B. J.,** Leng, Y., Wu, D., Bushley, K. E., Ohm, R. A., Otillar, R., et al. (2013). Comparative genome structure, secondary metabolite, and effector coding capacity across _Cochliobolus_ pathogens. _PLoS Genetics_, _9_(1), e1003233. doi:10.1371/journal.pgen.1003233

Xue, C., Wu, D., **Condon, B.**, Bi, Q., Wang, W., & Turgeon, B. G. (2013). Efficient gene knockout in the maize pathogen _Setosphaeria turcica_ using _Agrobacterium tumefaciens_-mediated transformation. _Phytopathology_, _103_(6), 641–647. doi:10.1094/PHYTO-08-12-0199-R

Ohm, R. A., Feau, N., Henrissat, B., Schoch, C. L., Horwitz, B. A., Barry, K. W., **Condon, B. J.,** et al. (2012). Diverse lifestyles and strategies of plant pathogenesis encoded in the genomes of eighteen Dothideomycetes fungi. _PLoS Pathogens_, _8_(12), e1003037. doi:10.1371/journal.ppat.1003037

Turgeon, B. G., **Condon, B. J**., Liu, J., & Zhang, N. (2010). Protoplast transformation of filamentous fungi. _Methods in Molecular Biology (Clifton, N.J.)_, _638_(Chapter 1), 3–19. doi:10.1007/978-1-60761-611-5_1

Almeida, N. F., Yan, S., Lindeberg, M., Studholme, D. J., Schneider, D. J., **Condon, B. J.,** et al. (2009). A draft genome sequence of _Pseudomonas syringae_ pv. tomato T1 reveals a type III effector repertoire significantly divergent from that _of Pseudomonas syringae_ pv. tomato DC3000. _Molecular Plant-Microbe Interactions : MPMI_, _22_(1), 52–62. doi:10.1094/MPMI-22-1-0052
