<!DOCTYPE html>
<html lang="en">

<link href="http://gmpg.org/xfn/11" rel="profile">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta http-equiv="content-type" content="text/html; charset=utf-8">
<!-- Enable responsiveness on mobile devices-->
<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">
<head>
  <title>Bradford Condon, Developer and Bioinformaticist</title>
<meta charset="utf-8">
<meta name="author" content="Bradford Condon">
<meta name="description" content="Website, CV, and Blog of Bradford Condon.  Bradford is a web and mobile developer.  Bradford also holds a PhD in plant pathology and is an expert in genomics, bioinformatics.">
</head>
<section class="hero is-primary is-bold">
    <div class="hero-body">
        <div class="container">
            <h1 class="title">
                <a href="/" title="Home">
                    Bradford Condon PhD
                </a>
            </h1>
            <h2 class="subtitle">
                Bioinformatics, Web & Mobile Development
            </h2>
        </div>

    </div>
    <div class="hero-foot">
        <nav class="tabs is-boxed nav-menu has-shadow">
            <div class="nav-center">
                <ul>

                    <li >
                    <a href="/">
          <span class="fa-stack fa-lg">
            <i class="fa fa-square-o fa-stack-2x"></i>
            <i class="fa fa-home fa-stack-1x"></i>
          </span>
                        Home</a></li>
                    <li >

                    <a href="/blog/">

          <span class="fa-stack fa-lg">
            <i class="fa fa-square-o fa-stack-2x"></i>
            <i class="fa fa-newspaper-o fa-stack-1x"></i>
          </span>
                        Blog</a>
                    </li>
                    <li >

                    <a href="/about/">
                   <span class="fa-stack fa-lg">
                      <i class="fa fa-square-o fa-stack-2x"></i>
                      <i class="fa fa-question fa-stack-1x"></i>
                    </span>
                        About</a>

                    </li>
                    <li >

                    <a href="/cv.html">
                    <span class="fa-stack fa-lg">
                        <i class="fa fa-square-o fa-stack-2x"></i>
                        <i class="fa fa-id-card fa-stack-1x"></i>
                   </span>
                        CV</a></li>
                </ul>
            </div>
        </nav>
    </div>
</section>
<link rel="stylesheet" href="/assets/css/application.css" />
<link rel="stylesheet" href="/assets/css/prism.css" />
<link rel="apple-touch-icon-precomposed" sizes="144x144"
      href="/public/apple-touch-icon-precomposed.png">
<link rel="shortcut icon" href="/public/favicon.ico">
<link rel="alternate" type="application/atom+xml" title="Bradford Condon PhD" href="/atom.xml">

<br/>

<body>
<div class="container content">
    <main>
        <div>
  <div class="box">
<article class="post">
  <h1 class="post-title">Velvet assembly output files</h1>
 <time datetime="2017-06-13T00:00:00-04:00" class="post-date">13 Jun 2017</time>
  <div class="card-content">
  <p>This post is part 4 of a series on file formats, written for the 2017 UK-KBRIN Essentials of Next 
Generation Sequencing Workshop at the University of Kentucky.</p>

<h2 id="introduction">Introduction</h2>
<p>The velvet manual is hosted online in wiki format.  You can find the section on output files <a href="https://github.com/dzerbino/velvet/wiki/Manual#output-files">here</a>.  I’ll be including relevant quotes from the manual.  I also found <a href="http://davetang.org/wiki/tiki-index.php?page=Velvet">this site</a> helpful, and some of the examples are taken from here.
Ultimately the stats files included with the assembly can be more confusing than helpful.  Don’t be too concerned if you don’t understand all of the metrics.</p>

<p>Below is a brief guide to the files included in the output.</p>

<h2 id="contigsfa">contigs.fa</h2>

<p>From the manual:</p>

<blockquote>
  <p>This fasta file contains the sequences of the contigs longer than 2k, where k is the word-length used in velveth. If you have specified a min_contig_lgth threshold, then the contigs shorter than that value are omitted.</p>
</blockquote>

<h2 id="statstxt">stats.txt</h2>

<p>This file describes the nodes in the assembly and will look like the below example.</p>

<table>
  <thead>
    <tr>
      <th>ID</th>
      <th>lgth</th>
      <th>out</th>
      <th>in</th>
      <th>long_cov</th>
      <th>short1_cov</th>
      <th>short1_Ocov</th>
      <th>short2_cov</th>
      <th>short2_Ocov</th>
      <th>long_nb</th>
      <th>short1_nb</th>
      <th>short2_nb</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>1</td>
      <td>153</td>
      <td>0</td>
      <td>0</td>
      <td>0.000000</td>
      <td>118.790850</td>
      <td>118.790850</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0</td>
      <td>N/A</td>
      <td>N/A</td>
    </tr>
    <tr>
      <td>2</td>
      <td>55</td>
      <td>0</td>
      <td>0</td>
      <td>0.000000</td>
      <td>14.581818</td>
      <td>14.581818</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0</td>
      <td>N/A</td>
      <td>N/A</td>
    </tr>
  </tbody>
</table>

<p>A table describing the headers is below</p>

<table>
  <thead>
    <tr>
      <th>Column header</th>
      <th>Full name</th>
      <th>Description</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>ID</td>
      <td>ID</td>
      <td>Node ID</td>
    </tr>
    <tr>
      <td>lgth</td>
      <td>length</td>
      <td>length (in k-mers)</td>
    </tr>
    <tr>
      <td>out</td>
      <td>3’ arcs</td>
      <td>Number of arcs 3’</td>
    </tr>
    <tr>
      <td>in</td>
      <td>5’ arcs</td>
      <td>Number of arcs 5’</td>
    </tr>
    <tr>
      <td>long_cov</td>
      <td>coverage short 1</td>
      <td>coverage in long reads</td>
    </tr>
    <tr>
      <td>short1_cov</td>
      <td>coverage short 1</td>
      <td>coverage in short reads (including divergent sequences)</td>
    </tr>
    <tr>
      <td>short1_Ocov</td>
      <td>coverage short 2</td>
      <td>coverage in short reads (conform to consensus only, strict)</td>
    </tr>
    <tr>
      <td>short2_cov</td>
      <td>coverage short 2</td>
      <td>coverage in short reads (including divergent sequences)</td>
    </tr>
    <tr>
      <td>short2_Ocov</td>
      <td>coverage short 2</td>
      <td>coverage in short reads (conform to consensus only, strict)</td>
    </tr>
    <tr>
      <td>long_nb</td>
      <td>long reads in node</td>
      <td>number of long reads in node</td>
    </tr>
    <tr>
      <td>short1_nb</td>
      <td>short 1 reads in node</td>
      <td>number of short1 reads in node</td>
    </tr>
    <tr>
      <td>short2_nb</td>
      <td>short 2 reads in node</td>
      <td>number of short2 reads in node</td>
    </tr>
  </tbody>
</table>

<h2 id="velvet_asmafg">velvet_asm.afg</h2>
<p>This file is meant to be read by the AMOS assembly package, so we will not be concerned with it.</p>

<h2 id="lastgraph">LastGraph</h2>

<p>This file describes the graph produced by velvet to create the assembly.</p>

<p>There is one header line for the entire graph, which lists the number of nodes, number of sequences, and the total hash length.</p>

<p>Then, each node has a block with the following format:</p>

<div class="highlighter-rouge"><pre class="highlight"><code> NODE $NODE_ID $COV_SHORT1 $O_COV_SHORT1 $COV_SHORT2 $O_COV_SHORT2 $ENDS_OF_KMERS_OF_NODE $ENDS_OF_KMERS_OF_TWIN_NODE 
</code></pre>
</div>

<p>These should look familiar to whats in the <code class="highlighter-rouge">stats.txt</code> file.  The ends of K-mers values are the last nucleotides of the k-mers in that node.</p>


  </div>
</article>
  </div>



 <div class="box">
<aside class="related">
  <h3>Related posts</h3>
  <ul class="related-posts">
    
      <li>
        <a href="/2018/01/17/d3_part3_nesting/">
          D3 part three: Nesting
          <small><time datetime="2018-01-17T00:00:00-05:00">17 Jan 2018</time></small>
        </a>
      </li>
    
      <li>
        <a href="/2018/01/12/gmod_day_1_2018/">
          Chado 1.4 at the 2018 GMOD hackathon
          <small><time datetime="2018-01-12T00:00:00-05:00">12 Jan 2018</time></small>
        </a>
      </li>
    
      <li>
        <a href="/2017/12/31/d3_part2_flipping/">
          D3 part two: flipping the plot
          <small><time datetime="2017-12-31T00:00:00-05:00">31 Dec 2017</time></small>
        </a>
      </li>
    
  </ul>
</aside>
 </div>


   
 </div>
    </main>
</div>

<footer class="footer">
      <div class="container">

      <div class="content has-text-centered">
      <span>
        <a href="https://github.com/bradfordcondon"><i class="fa fa-github-square fa-3x"></i></a>
        <a href="https://twitter.com/BradfordCondon"><i class="fa fa-twitter-square fa-3x"></i></a>
        <a href="https://www.linkedin.com/in/bradford-condon-0843823b/"><i class="fa fa-linkedin-square fa-3x"></i></a>

        </span>
        </div>
              <div class="content has-text-centered">

          <p>
          <time datetime="2018-01-17T21:56:26-05:00"> Bradford Condon 2018</time>. Site and content provided under a <a href="https://creativecommons.org/licenses/by-nc/2.0/"> Creative Commons non-commercial 2.0 license</a>.  Proudly hosted on Github Pages. 
          </p> 
          </div>
      
        </div>
        
      </footer>
<script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
            (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-71775012-1', 'auto');
    ga('send', 'pageview');

</script>
</body>
</html>
