<h1 id="goal">Goal</h1>

<p>The primary goal here is to create a track of synteny mappings for all genomes for jbrowse.  Any number of synteny calculations could work.</p>

<ul>
  <li>Mummer</li>
  <li>i-adhore 3.0</li>
</ul>

<h1 id="mummer">Mummer</h1>
<p>This module will allow converting nucmer coordinates to a gff file.  https://metacpan.org/pod/distribution/Bio-Gonzales/bin/coords2gff.pl</p>

<h1 id="i-adhore">i-adhore</h1>

<p>The software can be downloaded <a href="http://bioinformatics.psb.ugent.be/webtools/i-adhore/licensing/">here</a>.</p>

<p>Reference</p>

<blockquote>
  <p>Proost, S. et al. i-ADHoRe 3.0—fast and sensitive detection of genomic homology in extremely large data sets. Nucl. Acids Res. 40, e11–e11 (2012).</p>
</blockquote>

<h2 id="installation-and-set-up">Installation and set up</h2>

<div class="highlighter-rouge"><pre class="highlight"><code>tar -xzvf i-adhore-xxx.tar.gz
cd i-adhore-xxx

mkdir build
cd build
cmake .. -DCMAKE_INSTALL_PREFIX=usr/local/bin/i-adhore
make
sudo make install
</code></pre>
</div>

<p>Unfortunately make fails</p>

<div class="highlighter-rouge"><pre class="highlight"><code>
[  9%] Building CXX object src/alignment/CMakeFiles/alignment.dir/GGAligner.cpp.o
In file included from /Users/chet/Downloads/i-adhore-3.0.01/src/alignment/GGAligner.cpp:1:
In file included from /Users/chet/Downloads/i-adhore-3.0.01/src/alignment/GGAligner.h:6:
In file included from /Users/chet/Downloads/i-adhore-3.0.01/src/alignment/Aligner.h:5:
In file included from /Users/chet/Downloads/i-adhore-3.0.01/src/alignment/../GeneList.h:7:
In file included from /Users/chet/Downloads/i-adhore-3.0.01/src/alignment/../headers.h:30:
/Users/chet/Downloads/i-adhore-3.0.01/src/alignment/../datastructures/stringhash.h:15:11: error:
      use of undeclared identifier '__stl_hash_string'
        { return __stl_hash_string(s.c_str()); }
                         ^
In file included from /Users/chet/Downloads/i-adhore-3.0.01/src/alignment/GGAligner.cpp:1:
In file included from /Users/chet/Downloads/i-adhore-3.0.01/src/alignment/GGAligner.h:6:
In file included from /Users/chet/Downloads/i-adhore-3.0.01/src/alignment/Aligner.h:6:

</code></pre>
</div>
<p>Possible solution: https://github.com/Kentzo/armv6-rpi-linux-gnueabihf/issues/5</p>

<p><code class="highlighter-rouge">brew install gcc</code> will be very slow, so instead we try <code class="highlighter-rouge">brew install gcc48</code>.</p>

<p>if this doenst ork
need to tell cmake to use a specific compiler</p>

<div class="highlighter-rouge"><pre class="highlight"><code>export CC=/usr/local/bin/gcc
export CXX=/usr/local/bin/g++ ##  4.8 wahtever , find path
export CXX=/usr/local/Cellar/gcc@4.8/4.8.5
cmake /path/to/your/project
make
</code></pre>
</div>
<p><code class="highlighter-rouge">/usr/local/Cellar/gcc@4.8/4.8.5</code></p>

<h2 id="other-solution-from-neil">Other solution from Neil</h2>

<p><code class="highlighter-rouge">/Users/chet/Downloads/i-adhore-3.0.01/src/alignment/../datastructures/stringhash.h:15:11: error:</code></p>

<p>In the above file,
change <code class="highlighter-rouge">__stl_hash_string(s.c_str())</code> to <code class="highlighter-rouge">std::hash(s.c_str())</code></p>

<p>We create a script that runs i-adhore with a new tmp directory: <code class="highlighter-rouge">TMPDIR=/tmp i-adhore</code>.  The script also must pass arguments to the command.</p>

<div class="highlighter-rouge"><pre class="highlight"><code><span class="c">#! /bin/sh</span>
<span class="nv">TMPDIR</span><span class="o">=</span>/tmp
<span class="nb">exec</span> /usr/local/bin/i-adhore_3.0/bin/i-adhore  <span class="nv">$1</span>
</code></pre>
</div>

<h2 id="sibelia">Sibelia</h2>

<p><a href="http://bioinf.spbau.ru/sibelia">Project page</a></p>

